{
  "version": "15.0.6",
  "vulnerabilities": [
    {
      "id": "f00fea3f381f4f5f9b1f7c0ae44e07f8c65c529470bf764f49dfe8f4400b4364",
      "category": "container_scanning",
      "message": "CVE-2021-3487 on alpine/binutils@2.33.1-r0",
      "cve": "CVE-2021-3487",
      "severity": "Unknown",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/binutils@2.33.1-r0 to 2.33.1-r1",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/binutils@2.33.1-r0?os_name=alpine&os_version=3.11"
          },
          "version": "2.33.1-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-3487",
          "value": "CVE-2021-3487",
          "url": "https://scout.docker.com/v/CVE-2021-3487?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C2.33.1-r1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-3487?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C2.33.1-r1"
        }
      ]
    },
    {
      "id": "a5b4671b3ea59a9be9f3b7112a9e625b524de04223baaf93437cb1c0ef7d4000",
      "category": "container_scanning",
      "message": "GHSA-5cpq-8wj7-hf2v on cryptography@2.9.2",
      "description": "pyca/cryptography's wheels include a statically linked copy of OpenSSL. The versions of OpenSSL included in cryptography 0.5-40.0.2 are vulnerable to a security issue. More details about the vulnerability itself can be found in https://www.openssl.org/news/secadv/20230530.txt.\n\nIf you are building cryptography source (\"sdist\") then you are responsible for upgrading your copy of OpenSSL. Only users installing from wheels built by the cryptography project (i.e., those distributed on PyPI) need to update their cryptography versions.",
      "cve": "GHSA-5cpq-8wj7-hf2v",
      "severity": "Low",
      "confidence": "Unknown",
      "solution": "Upgrade cryptography@2.9.2 to 41.0.0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/cryptography@2.9.2"
          },
          "version": "2.9.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "GHSA-5cpq-8wj7-hf2v",
          "value": "GHSA-5cpq-8wj7-hf2v",
          "url": "https://scout.docker.com/v/GHSA-5cpq-8wj7-hf2v?s=github&n=cryptography&t=pypi&vr=%3E%3D0.5%2C%3C%3D40.0.2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/GHSA-5cpq-8wj7-hf2v?s=github&n=cryptography&t=pypi&vr=%3E%3D0.5%2C%3C%3D40.0.2"
        }
      ]
    },
    {
      "id": "c488b8cb2971656ca1337fa2aa2d4ddef17d8080007c19c140881b1e32262e94",
      "category": "container_scanning",
      "message": "GHSA-jm77-qphf-c4w8 on cryptography@2.9.2",
      "description": "pyca/cryptography's wheels include a statically linked copy of OpenSSL. The versions of OpenSSL included in cryptography 0.8-41.0.2 are vulnerable to several security issues. More details about the vulnerabilities themselves can be found in https://www.openssl.org/news/secadv/20230731.txt, https://www.openssl.org/news/secadv/20230719.txt, and https://www.openssl.org/news/secadv/20230714.txt.\n\nIf you are building cryptography source (\"sdist\") then you are responsible for upgrading your copy of OpenSSL. Only users installing from wheels built by the cryptography project (i.e., those distributed on PyPI) need to update their cryptography versions.",
      "cve": "GHSA-jm77-qphf-c4w8",
      "severity": "Low",
      "confidence": "Unknown",
      "solution": "Upgrade cryptography@2.9.2 to 41.0.3",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/cryptography@2.9.2"
          },
          "version": "2.9.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "GHSA-jm77-qphf-c4w8",
          "value": "GHSA-jm77-qphf-c4w8",
          "url": "https://scout.docker.com/v/GHSA-jm77-qphf-c4w8?s=github&n=cryptography&t=pypi&vr=%3E%3D0.8%2C%3C41.0.3"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/GHSA-jm77-qphf-c4w8?s=github&n=cryptography&t=pypi&vr=%3E%3D0.8%2C%3C41.0.3"
        }
      ]
    },
    {
      "id": "c741917070eb39d50aa92f4bb51192cbb33ce4572565b8e7582756fdf11d9bef",
      "category": "container_scanning",
      "message": "GHSA-v8gr-m533-ghj9 on cryptography@2.9.2",
      "description": "pyca/cryptography's wheels include a statically linked copy of OpenSSL. The versions of OpenSSL included in cryptography 2.5-41.0.3 are vulnerable to several security issues. More details about the vulnerabilities themselves can be found in https://www.openssl.org/news/secadv/20230908.txt.\n\nIf you are building cryptography source (\"sdist\") then you are responsible for upgrading your copy of OpenSSL. Only users installing from wheels built by the cryptography project (i.e., those distributed on PyPI) need to update their cryptography versions.",
      "cve": "GHSA-v8gr-m533-ghj9",
      "severity": "Low",
      "confidence": "Unknown",
      "solution": "Upgrade cryptography@2.9.2 to 41.0.4",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/cryptography@2.9.2"
          },
          "version": "2.9.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "GHSA-v8gr-m533-ghj9",
          "value": "GHSA-v8gr-m533-ghj9",
          "url": "https://scout.docker.com/v/GHSA-v8gr-m533-ghj9?s=github&n=cryptography&t=pypi&vr=%3E%3D2.5%2C%3C41.0.4"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/GHSA-v8gr-m533-ghj9?s=github&n=cryptography&t=pypi&vr=%3E%3D2.5%2C%3C41.0.4"
        }
      ]
    },
    {
      "id": "5781b0780df44d69ac2dc84ae0f42d594b8c41d9b612581451011dc156dd9aff",
      "category": "container_scanning",
      "message": "CVE-2021-23839 on alpine/openssl@1.1.1g-r0",
      "cve": "CVE-2021-23839",
      "severity": "Low",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/openssl@1.1.1g-r0 to 1.1.1j-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/openssl@1.1.1g-r0?os_name=alpine&os_version=3.11"
          },
          "version": "1.1.1g-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-23839",
          "value": "CVE-2021-23839",
          "url": "https://scout.docker.com/v/CVE-2021-23839?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1j-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-23839?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1j-r0"
        }
      ]
    },
    {
      "id": "18124f1eff81d0aa73f0959398801b40373f4703828915df3c88eb74d366a067",
      "category": "container_scanning",
      "message": "CVE-2022-22935 on salt@2019.2.3",
      "description": "An issue was discovered in SaltStack Salt in versions before 3002.8, 3003.4, 3004.1. A minion authentication denial of service can cause a MiTM attacker to force a minion process to stop by impersonating a master.",
      "cve": "CVE-2022-22935",
      "severity": "Low",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3002.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2022-22935",
          "value": "CVE-2022-22935",
          "url": "https://scout.docker.com/v/CVE-2022-22935?s=github&n=salt&t=pypi&vr=%3C3002.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2022-22935?s=github&n=salt&t=pypi&vr=%3C3002.8"
        }
      ]
    },
    {
      "id": "813b712fd50a27256c3025cbfe9191a5b11b1aa8366d70e3f4cc584f32947527",
      "category": "container_scanning",
      "message": "GHSA-qppv-j76h-2rpx on tornado@4.5.3",
      "description": "## Summary\nTornado interprets `-`, `+`, and `_` in chunk length and `Content-Length` values, which are not allowed by the HTTP RFCs. This can result in request smuggling when Tornado is deployed behind certain proxies that interpret those non-standard characters differently. This is known to apply to older versions of haproxy, although the current release is not affected.\n\n## Details\nTornado uses the `int` constructor to parse the values of `Content-Length` headers and chunk lengths in the following locations:\n### `tornado/http1connection.py:445`\n```python3\n            self._expected_content_remaining = int(headers[\"Content-Length\"])\n```\n### `tornado/http1connection.py:621`\n```python3\n                content_length = int(headers[\"Content-Length\"])  # type: Optional[int]\n```\n### `tornado/http1connection.py:671`\n```python3\n            chunk_len = int(chunk_len_str.strip(), 16)\n```\nBecause `int(\"0_0\") == int(\"+0\") == int(\"-0\") == int(\"0\")`, using the `int` constructor to parse and validate strings that should contain only ASCII digits is not a good strategy. \n\n",
      "cve": "GHSA-qppv-j76h-2rpx",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade tornado@4.5.3 to 6.3.3",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/tornado@4.5.3"
          },
          "version": "4.5.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "GHSA-qppv-j76h-2rpx",
          "value": "GHSA-qppv-j76h-2rpx",
          "url": "https://scout.docker.com/v/GHSA-qppv-j76h-2rpx?s=github&n=tornado&t=pypi&vr=%3C6.3.3"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/GHSA-qppv-j76h-2rpx?s=github&n=tornado&t=pypi&vr=%3C6.3.3"
        }
      ]
    },
    {
      "id": "31c944eae8103293556a16a0e2b7078db6ec3bac1833cb1d38c093435cbfdf47",
      "category": "container_scanning",
      "message": "CVE-2023-20898 on salt@2019.2.3",
      "description": "Git Providers can read from the wrong environment because they get the same cache directory base name in Salt masters prior to 3005.2 or 3006.2. Anything that uses Git Providers with different environments can get garbage data or the wrong data, which can lead to wrongful data disclosure, wrongful executions, data corruption and/or crash.",
      "cve": "CVE-2023-20898",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3005.2",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2023-20898",
          "value": "CVE-2023-20898",
          "url": "https://scout.docker.com/v/CVE-2023-20898?s=github&n=salt&t=pypi&vr=%3C3005.2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2023-20898?s=github&n=salt&t=pypi&vr=%3C3005.2"
        }
      ]
    },
    {
      "id": "b7228816b98cfa8856b6c912468434c67a1daa7f30cc380f9de8428206498dea",
      "category": "container_scanning",
      "message": "CVE-2021-25284 on salt@2019.2.3",
      "description": "An issue was discovered in through SaltStack Salt before 3002.5. `salt.modules.cmdmod` can log credentials to the info or error log level.",
      "cve": "CVE-2021-25284",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-25284",
          "value": "CVE-2021-25284",
          "url": "https://scout.docker.com/v/CVE-2021-25284?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-25284?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ]
    },
    {
      "id": "352e31406082f3153db4daf8d3b3c2e92d1ec5cc3cd3df1d2317f8fef384a43d",
      "category": "container_scanning",
      "message": "CVE-2024-37891 on urllib3@1.25.9",
      "description": "When using urllib3's proxy support with `ProxyManager`, the `Proxy-Authorization` header is only sent to the configured proxy, as expected.\n\nHowever, when sending HTTP requests *without* using urllib3's proxy support, it's possible to accidentally configure the `Proxy-Authorization` header even though it won't have any effect as the request is not using a forwarding proxy or a tunneling proxy. In those cases, urllib3 doesn't treat the `Proxy-Authorization` HTTP header as one carrying authentication material and thus doesn't strip the header on cross-origin redirects.\n\nBecause this is a highly unlikely scenario, we believe the severity of this vulnerability is low for almost all users. Out of an abundance of caution urllib3 will automatically strip the `Proxy-Authorization` header during cross-origin redirects to avoid the small chance that users are doing this on accident.\n\nUsers should use urllib3's proxy support or disable automatic redirects to achieve safe processing of the `Proxy-Authorization` header, but we still decided to strip the header by default in order to further protect users who aren't using the correct approach.\n\n## Affected usages\n\nWe believe the number of usages affected by this advisory is low. It requires all of the following to be true to be exploited:\n\n* Setting the `Proxy-Authorization` header without using urllib3's built-in proxy support.\n* Not disabling HTTP redirects.\n* Either not using an HTTPS origin server or for the proxy or target origin to redirect to a malicious origin.\n\n## Remediation\n\n* Using the `Proxy-Authorization` header with urllib3's `ProxyManager`.\n* Disabling HTTP redirects using `redirects=False` when sending requests.\n* Not using the `Proxy-Authorization` header.",
      "cve": "CVE-2024-37891",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade urllib3@1.25.9 to 1.26.19",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/urllib3@1.25.9"
          },
          "version": "1.25.9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2024-37891",
          "value": "CVE-2024-37891",
          "url": "https://scout.docker.com/v/CVE-2024-37891?s=github&n=urllib3&t=pypi&vr=%3C1.26.19"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2024-37891?s=github&n=urllib3&t=pypi&vr=%3C1.26.19"
        }
      ]
    },
    {
      "id": "2b1377fac277cf569eff13b89d33d7731d443d49c38423e2bb8f278460b75b1d",
      "category": "container_scanning",
      "message": "CVE-2024-22231 on salt@2019.2.3",
      "description": "Syndic cache directory creation is vulnerable to a directory traversal attack in salt project which can lead a malicious attacker to create an arbitrary directory on a Salt master.",
      "cve": "CVE-2024-22231",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3005.5",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2024-22231",
          "value": "CVE-2024-22231",
          "url": "https://scout.docker.com/v/CVE-2024-22231?s=github&n=salt&t=pypi&vr=%3C3005.5"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2024-22231?s=github&n=salt&t=pypi&vr=%3C3005.5"
        }
      ]
    },
    {
      "id": "ef50eec891d9221907cbfd6161ff2942353ae696764b46221a07b8a511dc7576",
      "category": "container_scanning",
      "message": "CVE-2019-17594 on alpine/ncurses@6.1_p20200118-r3",
      "cve": "CVE-2019-17594",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/ncurses@6.1_p20200118-r3?os_name=alpine&os_version=3.11"
          },
          "version": "6.1_p20200118-r3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-17594",
          "value": "CVE-2019-17594",
          "url": "https://scout.docker.com/v/CVE-2019-17594?s=alpine&n=ncurses&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D6.1_p20200118-r4"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-17594?s=alpine&n=ncurses&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D6.1_p20200118-r4"
        }
      ]
    },
    {
      "id": "b8ba97d65b3031996612e689ad90bd676806f1b15d10a4f9ce619247a7b071f6",
      "category": "container_scanning",
      "message": "CVE-2019-19924 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2019-19924",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-19924",
          "value": "CVE-2019-19924",
          "url": "https://scout.docker.com/v/CVE-2019-19924?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-19924?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "82e3ecad18a69b5540b3a3bbd067eaa2c0b109bfcc6781b0f788b64d079aab97",
      "category": "container_scanning",
      "message": "CVE-2021-42374 on alpine/busybox@1.31.1-r9",
      "cve": "CVE-2021-42374",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/busybox@1.31.1-r9 to 1.31.1-r11",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/busybox@1.31.1-r9?os_name=alpine&os_version=3.11"
          },
          "version": "1.31.1-r9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-42374",
          "value": "CVE-2021-42374",
          "url": "https://scout.docker.com/v/CVE-2021-42374?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-42374?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ]
    },
    {
      "id": "211275980ab15f3dd1f7cd60d1ccb0c13d872720a501c2abdb732fc1b4864201",
      "category": "container_scanning",
      "message": "CVE-2023-20897 on salt@2019.2.3",
      "description": "Salt masters prior to 3005.2 or 3006.2 contain a DOS in minion return. After receiving several bad packets on the request server equal to the number of worker threads, the master will become unresponsive to return requests until restarted.",
      "cve": "CVE-2023-20897",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3005.2",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2023-20897",
          "value": "CVE-2023-20897",
          "url": "https://scout.docker.com/v/CVE-2023-20897?s=github&n=salt&t=pypi&vr=%3C3005.2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2023-20897?s=github&n=salt&t=pypi&vr=%3C3005.2"
        }
      ]
    },
    {
      "id": "0c559f03b6bc6b4fc7bb7513beb4f3ad11e28325b0a5a4f08bc30059c07cddee",
      "category": "container_scanning",
      "message": "CVE-2023-28370 on tornado@4.5.3",
      "description": "Open redirect vulnerability in Tornado versions 6.3.1 and earlier allows a remote unauthenticated attacker to redirect a user to an arbitrary web site and conduct a phishing attack by having user access a specially crafted URL.",
      "cve": "CVE-2023-28370",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade tornado@4.5.3 to 6.3.2",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/tornado@4.5.3"
          },
          "version": "4.5.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2023-28370",
          "value": "CVE-2023-28370",
          "url": "https://scout.docker.com/v/CVE-2023-28370?s=github&n=tornado&t=pypi&vr=%3C6.3.2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2023-28370?s=github&n=tornado&t=pypi&vr=%3C6.3.2"
        }
      ]
    },
    {
      "id": "ab3fb093a66b3d0aa13d2b0300c469b0a8707ac79648fa06b3ca996d6388662c",
      "category": "container_scanning",
      "message": "CVE-2024-47081 on requests@2.23.0",
      "description": "### Impact\n\nDue to a URL parsing issue, Requests releases prior to 2.32.4 may leak .netrc credentials to third parties for specific maliciously-crafted URLs.\n\n### Workarounds\nFor older versions of Requests, use of the .netrc file can be disabled with `trust_env=False` on your Requests Session ([docs](https://requests.readthedocs.io/en/latest/api/#requests.Session.trust_env)).\n\n### References\nhttps://github.com/psf/requests/pull/6965\nhttps://seclists.org/fulldisclosure/2025/Jun/2",
      "cve": "CVE-2024-47081",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade requests@2.23.0 to 2.32.4",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/requests@2.23.0"
          },
          "version": "2.23.0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2024-47081",
          "value": "CVE-2024-47081",
          "url": "https://scout.docker.com/v/CVE-2024-47081?s=github&n=requests&t=pypi&vr=%3C2.32.4"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2024-47081?s=github&n=requests&t=pypi&vr=%3C2.32.4"
        }
      ]
    },
    {
      "id": "7db1cacb3c95a37c3e083c0c3af72b5a59319de49606ddd37e540336c434ab23",
      "category": "container_scanning",
      "message": "CVE-2025-50181 on urllib3@1.25.9",
      "description": "urllib3 handles redirects and retries using the same mechanism, which is controlled by the `Retry` object. The most common way to disable redirects is at the request level, as follows:\n\n```python\nresp = urllib3.request(\"GET\", \"https://httpbin.org/redirect/1\", redirect=False)\nprint(resp.status)\n# 302\n```\n\nHowever, it is also possible to disable redirects, for all requests, by instantiating a `PoolManager` and specifying `retries` in a way that disable redirects:\n\n```python\nimport urllib3\n\nhttp = urllib3.PoolManager(retries=0)  # should raise MaxRetryError on redirect\nhttp = urllib3.PoolManager(retries=urllib3.Retry(redirect=0))  # equivalent to the above\nhttp = urllib3.PoolManager(retries=False)  # should return the first response\n\nresp = http.request(\"GET\", \"https://httpbin.org/redirect/1\")\n```\n\nHowever, the `retries` parameter is currently ignored, which means all the above examples don't disable redirects.\n\n## Affected usages\n\nPassing `retries` on `PoolManager` instantiation to disable redirects or restrict their number.\n\nBy default, requests and botocore users are not affected.\n\n## Impact\n\nRedirects are often used to exploit SSRF vulnerabilities. An application attempting to mitigate SSRF or open redirect vulnerabilities by disabling redirects at the PoolManager level will remain vulnerable.\n\n## Remediation\n\nYou can remediate this vulnerability with the following steps:\n\n * Upgrade to a patched version of urllib3. If your organization would benefit from the continued support of urllib3 1.x, please contact [sethmichaellarson@gmail.com](mailto:sethmichaellarson@gmail.com) to discuss sponsorship or contribution opportunities.\n * Disable redirects at the `request()` level instead of the `PoolManager()` level.",
      "cve": "CVE-2025-50181",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade urllib3@1.25.9 to 2.5.0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/urllib3@1.25.9"
          },
          "version": "1.25.9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2025-50181",
          "value": "CVE-2025-50181",
          "url": "https://scout.docker.com/v/CVE-2025-50181?s=github&n=urllib3&t=pypi&vr=%3C2.5.0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2025-50181?s=github&n=urllib3&t=pypi&vr=%3C2.5.0"
        }
      ]
    },
    {
      "id": "96552d085768d3b532db3952a42897aa1566ec2ecf7885cd1828db1892b9a08a",
      "category": "container_scanning",
      "message": "GHSA-753j-mpmx-qq6g on tornado@4.5.3",
      "description": "### Summary\nWhen Tornado receives a request with two `Transfer-Encoding: chunked` headers, it ignores them both. This enables request smuggling when Tornado is deployed behind a proxy server that emits such requests. [Pound](https://en.wikipedia.org/wiki/Pound_(networking)) does this.\n\n### PoC\n0. Install Tornado.\n1. Start a simple Tornado server that echoes each received request's body:\n```bash\ncat << EOF > server.py\nimport asyncio\nimport tornado\n\nclass MainHandler(tornado.web.RequestHandler):\n    def post(self):\n        self.write(self.request.body)\n\nasync def main():\n    tornado.web.Application([(r\"/\", MainHandler)]).listen(8000)\n    await asyncio.Event().wait()\n\nasyncio.run(main())\nEOF\npython3 server.py &\n```\n2. Send a valid chunked request:\n```bash\nprintf 'POST / HTTP/1.1\\r\\nTransfer-Encoding: chunked\\r\\n\\r\\n1\\r\\nZ\\r\\n0\\r\\n\\r\\n' | nc localhost 8000\n```\n3. Observe that the response is as expected:\n```\nHTTP/1.1 200 OK\nServer: TornadoServer/6.3.3\nContent-Type: text/html; charset=UTF-8\nDate: Sat, 07 Oct 2023 17:32:05 GMT\nContent-Length: 1\n\nZ\n```\n4. Send a request with two `Transfer-Encoding: chunked` headers:\n```\nprintf 'POST / HTTP/1.1\\r\\nTransfer-Encoding: chunked\\r\\nTransfer-Encoding: chunked\\r\\n\\r\\n1\\r\\nZ\\r\\n0\\r\\n\\r\\n' | nc localhost 8000\n```\n5. Observe the strange response:\n```\nHTTP/1.1 200 OK\nServer: TornadoServer/6.3.3\nContent-Type: text/html; charset=UTF-8\nDate: Sat, 07 Oct 2023 17:35:40 GMT\nContent-Length: 0\n\nHTTP/1.1 400 Bad Request\n\n```\nThis is because Tornado believes that the request has no message body, so it tries to interpret `1\\r\\nZ\\r\\n0\\r\\n\\r\\n` as its own request, which causes a 400 response. With a little cleverness involving `chunk-ext`s, you can get Tornado to instead respond 405, which has the potential to desynchronize the connection, as opposed to 400 which should always result in a connection closure.\n\n### Impact\nAnyone using Tornado behind a proxy that forwards requests containing multiple `Transfer-Encoding: chunked` headers is vulnerable to request smuggling, which may entail ACL bypass, cache poisoning, or connection desynchronization.\n",
      "cve": "GHSA-753j-mpmx-qq6g",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade tornado@4.5.3 to 6.4.1",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/tornado@4.5.3"
          },
          "version": "4.5.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "GHSA-753j-mpmx-qq6g",
          "value": "GHSA-753j-mpmx-qq6g",
          "url": "https://scout.docker.com/v/GHSA-753j-mpmx-qq6g?s=github&n=tornado&t=pypi&vr=%3C%3D6.4.0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/GHSA-753j-mpmx-qq6g?s=github&n=tornado&t=pypi&vr=%3C%3D6.4.0"
        }
      ]
    },
    {
      "id": "4aef3bfddd059a49a2df7cdb03a45608b012962ebb9ea3afec80e7a991c1f1a7",
      "category": "container_scanning",
      "message": "CVE-2019-17595 on alpine/ncurses@6.1_p20200118-r3",
      "cve": "CVE-2019-17595",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/ncurses@6.1_p20200118-r3?os_name=alpine&os_version=3.11"
          },
          "version": "6.1_p20200118-r3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-17595",
          "value": "CVE-2019-17595",
          "url": "https://scout.docker.com/v/CVE-2019-17595?s=alpine&n=ncurses&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D6.1_p20200118-r4"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-17595?s=alpine&n=ncurses&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D6.1_p20200118-r4"
        }
      ]
    },
    {
      "id": "66f32f601a42c74205105aff5b1a7c27b9a2f80ea4b7047fbcb263ccda30530c",
      "category": "container_scanning",
      "message": "CVE-2024-22195 on jinja2@2.11.2",
      "description": "The `xmlattr` filter in affected versions of Jinja accepts keys containing spaces. XML/HTML attributes cannot contain spaces, as each would then be interpreted as a separate attribute. If an application accepts keys (as opposed to only values) as user input, and renders these in pages that other users see as well, an attacker could use this to inject other attributes and perform XSS. Note that accepting keys as user input is not common or a particularly intended use case of the `xmlattr` filter, and an application doing so should already be verifying what keys are provided regardless of this fix.",
      "cve": "CVE-2024-22195",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade jinja2@2.11.2 to 3.1.3",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/jinja2@2.11.2"
          },
          "version": "2.11.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2024-22195",
          "value": "CVE-2024-22195",
          "url": "https://scout.docker.com/v/CVE-2024-22195?s=github&n=jinja2&t=pypi&vr=%3C3.1.3"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2024-22195?s=github&n=jinja2&t=pypi&vr=%3C3.1.3"
        }
      ]
    },
    {
      "id": "8bad44d4309c713d6cca0ff9b5fb5772b2ee8f9d97b7a68bb724e0f99ecaf526",
      "category": "container_scanning",
      "message": "CVE-2024-34064 on jinja2@2.11.2",
      "description": "The `xmlattr` filter in affected versions of Jinja accepts keys containing non-attribute characters. XML/HTML attributes cannot contain spaces, `/`, `>`, or `=`, as each would then be interpreted as starting a separate attribute. If an application accepts keys (as opposed to only values) as user input, and renders these in pages that other users see as well, an attacker could use this to inject other attributes and perform XSS. The fix for the previous GHSA-h5c8-rqwp-cp95 CVE-2024-22195 only addressed spaces but not other characters.\n\nAccepting keys as user input is now explicitly considered an unintended use case of the `xmlattr` filter, and code that does so without otherwise validating the input should be flagged as insecure, regardless of Jinja version. Accepting _values_ as user input continues to be safe.",
      "cve": "CVE-2024-34064",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade jinja2@2.11.2 to 3.1.4",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/jinja2@2.11.2"
          },
          "version": "2.11.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2024-34064",
          "value": "CVE-2024-34064",
          "url": "https://scout.docker.com/v/CVE-2024-34064?s=github&n=jinja2&t=pypi&vr=%3C3.1.4"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2024-34064?s=github&n=jinja2&t=pypi&vr=%3C3.1.4"
        }
      ]
    },
    {
      "id": "b46e26856a792eb45dec7f2a5b2914eb9fb1af9dc2e6e5ae0fa0bce09e8c44f7",
      "category": "container_scanning",
      "message": "CVE-2024-56326 on jinja2@2.11.2",
      "description": "An oversight in how the Jinja sandboxed environment detects calls to `str.format` allows an attacker that controls the content of a template to execute arbitrary Python code.\n\nTo exploit the vulnerability, an attacker needs to control the content of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates.\n\nJinja's sandbox does catch calls to `str.format` and ensures they don't escape the sandbox. However, it's possible to store a reference to a malicious string's `format` method, then pass that to a filter that calls it. No such filters are built-in to Jinja, but could be present through custom filters in an application. After the fix, such indirect calls are also handled by the sandbox.",
      "cve": "CVE-2024-56326",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade jinja2@2.11.2 to 3.1.5",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/jinja2@2.11.2"
          },
          "version": "2.11.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2024-56326",
          "value": "CVE-2024-56326",
          "url": "https://scout.docker.com/v/CVE-2024-56326?s=github&n=jinja2&t=pypi&vr=%3C%3D3.1.4"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2024-56326?s=github&n=jinja2&t=pypi&vr=%3C%3D3.1.4"
        }
      ]
    },
    {
      "id": "b94b4ece3a19a5c2f4780f95b038e6c41ccad7178222d2ba61365af15587d735",
      "category": "container_scanning",
      "message": "CVE-2025-27516 on jinja2@2.11.2",
      "description": "An oversight in how the Jinja sandboxed environment interacts with the `|attr` filter allows an attacker that controls the content of a template to execute arbitrary Python code.\n\nTo exploit the vulnerability, an attacker needs to control the content of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates.\n\nJinja's sandbox does catch calls to `str.format` and ensures they don't escape the sandbox. However, it's possible to use the `|attr` filter to get a reference to a string's plain format method, bypassing the sandbox. After the fix, the `|attr` filter no longer bypasses the environment's attribute lookup.",
      "cve": "CVE-2025-27516",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade jinja2@2.11.2 to 3.1.6",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/jinja2@2.11.2"
          },
          "version": "2.11.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2025-27516",
          "value": "CVE-2025-27516",
          "url": "https://scout.docker.com/v/CVE-2025-27516?s=github&n=jinja2&t=pypi&vr=%3C%3D3.1.5"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2025-27516?s=github&n=jinja2&t=pypi&vr=%3C%3D3.1.5"
        }
      ]
    },
    {
      "id": "b553c01e71f16936ccc9a9423cd52517d478104a47e8135d416de13f603d3662",
      "category": "container_scanning",
      "message": "CVE-2019-19645 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2019-19645",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-19645",
          "value": "CVE-2019-19645",
          "url": "https://scout.docker.com/v/CVE-2019-19645?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-19645?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "985c78a543d0e3e451278e3e58f27772694f353edc8bd5d0eaa70e436c3a1289",
      "category": "container_scanning",
      "message": "CVE-2020-13434 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2020-13434",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-13434",
          "value": "CVE-2020-13434",
          "url": "https://scout.docker.com/v/CVE-2020-13434?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-13434?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "a66d0af78dd02665cda141f8f5fec3e3c2140a39b97f9c7493105340657af9ef",
      "category": "container_scanning",
      "message": "CVE-2020-13435 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2020-13435",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-13435",
          "value": "CVE-2020-13435",
          "url": "https://scout.docker.com/v/CVE-2020-13435?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-13435?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "8341b9b0bb0f6865d1e677594ffd9c6f0e728741a3469186b48fb1144c63a20d",
      "category": "container_scanning",
      "message": "CVE-2020-13631 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2020-13631",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-13631",
          "value": "CVE-2020-13631",
          "url": "https://scout.docker.com/v/CVE-2020-13631?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-13631?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "7f5caeb6f6aae6a9f5aa6a0950ed6bf2edc4fca0fe7ffc26b8b51c136d546eaa",
      "category": "container_scanning",
      "message": "CVE-2020-13632 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2020-13632",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-13632",
          "value": "CVE-2020-13632",
          "url": "https://scout.docker.com/v/CVE-2020-13632?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-13632?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "7625ec0184bd4d6dbcfa44e3cd794e2d27ae909bdeb0602c54f679ab3e898580",
      "category": "container_scanning",
      "message": "CVE-2020-15358 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2020-15358",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-15358",
          "value": "CVE-2020-15358",
          "url": "https://scout.docker.com/v/CVE-2020-15358?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-15358?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "6600a0938dc7ed86851524e52e77572af35a6e34ebd718f9db75f3970cbacfc3",
      "category": "container_scanning",
      "message": "CVE-2020-17490 on salt@2019.2.3",
      "description": "The TLS module within SaltStack Salt through 3002 creates certificates with weak file permissions.",
      "cve": "CVE-2020-17490",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.6",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-17490",
          "value": "CVE-2020-17490",
          "url": "https://scout.docker.com/v/CVE-2020-17490?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.6"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-17490?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.6"
        }
      ]
    },
    {
      "id": "83f2677be138ff7cf142be9574b1907126a266a35f3bd4236d2907b592ec9992",
      "category": "container_scanning",
      "message": "CVE-2020-28928 on alpine/musl@1.1.24-r2",
      "cve": "CVE-2020-28928",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/musl@1.1.24-r2 to 1.1.24-r3",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/musl@1.1.24-r2?os_name=alpine&os_version=3.11"
          },
          "version": "1.1.24-r2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-28928",
          "value": "CVE-2020-28928",
          "url": "https://scout.docker.com/v/CVE-2020-28928?s=alpine&n=musl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.24-r3"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-28928?s=alpine&n=musl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.24-r3"
        }
      ]
    },
    {
      "id": "814876055807e7c7b014e8dc5cf706cb61db2ecd1ed58b5f6adbc7d1c9fac11f",
      "category": "container_scanning",
      "message": "CVE-2020-35493 on alpine/binutils@2.33.1-r0",
      "cve": "CVE-2020-35493",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/binutils@2.33.1-r0?os_name=alpine&os_version=3.11"
          },
          "version": "2.33.1-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-35493",
          "value": "CVE-2020-35493",
          "url": "https://scout.docker.com/v/CVE-2020-35493?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-35493?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ]
    },
    {
      "id": "a1c2ea8d576fd65f771bac48c552fa117dacfcefe5f240b7810e3c0cb36663f2",
      "category": "container_scanning",
      "message": "CVE-2020-35495 on alpine/binutils@2.33.1-r0",
      "cve": "CVE-2020-35495",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/binutils@2.33.1-r0?os_name=alpine&os_version=3.11"
          },
          "version": "2.33.1-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-35495",
          "value": "CVE-2020-35495",
          "url": "https://scout.docker.com/v/CVE-2020-35495?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-35495?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ]
    },
    {
      "id": "4b44960beb5b61b54ef1e35aa991474c8a729b8d0ccb45db8338b03286d888f1",
      "category": "container_scanning",
      "message": "CVE-2020-35496 on alpine/binutils@2.33.1-r0",
      "cve": "CVE-2020-35496",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/binutils@2.33.1-r0?os_name=alpine&os_version=3.11"
          },
          "version": "2.33.1-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-35496",
          "value": "CVE-2020-35496",
          "url": "https://scout.docker.com/v/CVE-2020-35496?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-35496?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ]
    },
    {
      "id": "f6328dc3841e6417f840ebd335be7a332515e2c278eec0d80d983b6ac085db1b",
      "category": "container_scanning",
      "message": "CVE-2020-35507 on alpine/binutils@2.33.1-r0",
      "cve": "CVE-2020-35507",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/binutils@2.33.1-r0?os_name=alpine&os_version=3.11"
          },
          "version": "2.33.1-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-35507",
          "value": "CVE-2020-35507",
          "url": "https://scout.docker.com/v/CVE-2020-35507?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-35507?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ]
    },
    {
      "id": "97ecaa72af217ffc4fedc808565b03c0b4bab4712e638b062b659f211858667b",
      "category": "container_scanning",
      "message": "CVE-2021-20227 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2021-20227",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-20227",
          "value": "CVE-2021-20227",
          "url": "https://scout.docker.com/v/CVE-2021-20227?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-20227?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "620caff992e91c24436870d7a761f233b7bc0c9932cfdd69c35df3cd7d5ad849",
      "category": "container_scanning",
      "message": "CVE-2021-37600 on alpine/util-linux@2.34-r1",
      "cve": "CVE-2021-37600",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/util-linux@2.34-r1?os_name=alpine&os_version=3.11"
          },
          "version": "2.34-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-37600",
          "value": "CVE-2021-37600",
          "url": "https://scout.docker.com/v/CVE-2021-37600?s=alpine&n=util-linux&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.34-r1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-37600?s=alpine&n=util-linux&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.34-r1"
        }
      ]
    },
    {
      "id": "49583e9e4966b930483ed02784b914051e04ff1b1f8590ca5df90333582e66de",
      "category": "container_scanning",
      "message": "CVE-2024-0727 on cryptography@2.9.2",
      "description": "Issue summary: Processing a maliciously formatted PKCS12 file may lead OpenSSL\nto crash leading to a potential Denial of Service attack\n\nImpact summary: Applications loading files in the PKCS12 format from untrusted\nsources might terminate abruptly.\n\nA file in PKCS12 format can contain certificates and keys and may come from an\nuntrusted source. The PKCS12 specification allows certain fields to be NULL, but\nOpenSSL does not correctly check for this case. This can lead to a NULL pointer\ndereference that results in OpenSSL crashing. If an application processes PKCS12\nfiles from an untrusted source using the OpenSSL APIs then that application will\nbe vulnerable to this issue.\n\nOpenSSL APIs that are vulnerable to this are: PKCS12_parse(),\nPKCS12_unpack_p7data(), PKCS12_unpack_p7encdata(), PKCS12_unpack_authsafes()\nand PKCS12_newpass().\n\nWe have also fixed a similar issue in SMIME_write_PKCS7(). However since this\nfunction is related to writing data we do not consider it security significant.\n\nThe FIPS modules in 3.2, 3.1 and 3.0 are not affected by this issue.",
      "cve": "CVE-2024-0727",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade cryptography@2.9.2 to 42.0.2",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/cryptography@2.9.2"
          },
          "version": "2.9.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2024-0727",
          "value": "CVE-2024-0727",
          "url": "https://scout.docker.com/v/CVE-2024-0727?s=github&n=cryptography&t=pypi&vr=%3C42.0.2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2024-0727?s=github&n=cryptography&t=pypi&vr=%3C42.0.2"
        }
      ]
    },
    {
      "id": "95dbb2fe0ead198321589c6213b59a08e720e62e8b370c9a1bc8c37aa2afea6c",
      "category": "container_scanning",
      "message": "CVE-2024-35195 on requests@2.23.0",
      "description": "When making requests through a Requests `Session`, if the first request is made with `verify=False` to disable cert verification, all subsequent requests to the same origin will continue to ignore cert verification regardless of changes to the value of `verify`. This behavior will continue for the lifecycle of the connection in the connection pool.\n\n### Remediation\nAny of these options can be used to remediate the current issue, we highly recommend upgrading as the preferred mitigation.\n\n* Upgrade to `requests>=2.32.0`.\n* For `requests<2.32.0`, avoid setting `verify=False` for the first request to a host while using a Requests Session.\n* For `requests<2.32.0`, call `close()` on `Session` objects to clear existing connections if `verify=False` is used.\n\n### Related Links\n* https://github.com/psf/requests/pull/6655",
      "cve": "CVE-2024-35195",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade requests@2.23.0 to 2.32.0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/requests@2.23.0"
          },
          "version": "2.23.0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2024-35195",
          "value": "CVE-2024-35195",
          "url": "https://scout.docker.com/v/CVE-2024-35195?s=github&n=requests&t=pypi&vr=%3C2.32.0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2024-35195?s=github&n=requests&t=pypi&vr=%3C2.32.0"
        }
      ]
    },
    {
      "id": "7e57c870d71de14e57952e4104a42339635ec72ac7c1b5704fd944a8f4046cf0",
      "category": "container_scanning",
      "message": "CVE-2023-45803 on urllib3@1.25.9",
      "description": "urllib3 previously wouldn't remove the HTTP request body when an HTTP redirect response using status 303 \"See Other\" after the request had its method changed from one that could accept a request body (like `POST`) to `GET` as is required by HTTP RFCs. Although the behavior of removing the request body is not specified in the section for redirects, it can be inferred by piecing together information from different sections and we have observed the behavior in other major HTTP client implementations like curl and web browsers.\n\nFrom [RFC 9110 Section 9.3.1](https://www.rfc-editor.org/rfc/rfc9110.html#name-get):\n\n> A client SHOULD NOT generate content in a GET request unless it is made directly to an origin server that has previously indicated, in or out of band, that such a request has a purpose and will be adequately supported.\n\n## Affected usages\n\nBecause the vulnerability requires a previously trusted service to become compromised in order to have an impact on confidentiality we believe the exploitability of this vulnerability is low. Additionally, many users aren't putting sensitive data in HTTP request bodies, if this is the case then this vulnerability isn't exploitable.\n\nBoth of the following conditions must be true to be affected by this vulnerability:\n\n* If you're using urllib3 and submitting sensitive information in the HTTP request body (such as form data or JSON)\n* The origin service is compromised and starts redirecting using 303 to a malicious peer or the redirected-to service becomes compromised.\n\n## Remediation\n\nYou can remediate this vulnerability with any of the following steps:\n\n* Upgrade to a patched version of urllib3 (v1.26.18 or v2.0.7)\n* Disable redirects for services that you aren't expecting to respond with redirects with `redirects=False`.\n* Disable automatic redirects with `redirects=False` and handle 303 redirects manually by stripping the HTTP request body.",
      "cve": "CVE-2023-45803",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade urllib3@1.25.9 to 1.26.18",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/urllib3@1.25.9"
          },
          "version": "1.25.9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2023-45803",
          "value": "CVE-2023-45803",
          "url": "https://scout.docker.com/v/CVE-2023-45803?s=github&n=urllib3&t=pypi&vr=%3C1.26.18"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2023-45803?s=github&n=urllib3&t=pypi&vr=%3C1.26.18"
        }
      ]
    },
    {
      "id": "98b3a761b34ff658f7e99514a06ea6b0fcdc9fcb187f5949714159a7ea7fc343",
      "category": "container_scanning",
      "message": "CVE-2020-1971 on alpine/openssl@1.1.1g-r0",
      "cve": "CVE-2020-1971",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/openssl@1.1.1g-r0 to 1.1.1i-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/openssl@1.1.1g-r0?os_name=alpine&os_version=3.11"
          },
          "version": "1.1.1g-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-1971",
          "value": "CVE-2020-1971",
          "url": "https://scout.docker.com/v/CVE-2020-1971?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1i-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-1971?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1i-r0"
        }
      ]
    },
    {
      "id": "eaf0a71bc8f8ec01a799e5b2d06f769f739353f2893223dcb55aea4b6a379e1b",
      "category": "container_scanning",
      "message": "CVE-2021-23841 on alpine/openssl@1.1.1g-r0",
      "cve": "CVE-2021-23841",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/openssl@1.1.1g-r0 to 1.1.1j-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/openssl@1.1.1g-r0?os_name=alpine&os_version=3.11"
          },
          "version": "1.1.1g-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-23841",
          "value": "CVE-2021-23841",
          "url": "https://scout.docker.com/v/CVE-2021-23841?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1j-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-23841?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1j-r0"
        }
      ]
    },
    {
      "id": "0f0ebec2957d5af4456aec1d4d77e4a96a823f4ac7cce4cb7096df0482bb6603",
      "category": "container_scanning",
      "message": "CVE-2021-3449 on alpine/openssl@1.1.1g-r0",
      "cve": "CVE-2021-3449",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/openssl@1.1.1g-r0 to 1.1.1k-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/openssl@1.1.1g-r0?os_name=alpine&os_version=3.11"
          },
          "version": "1.1.1g-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-3449",
          "value": "CVE-2021-3449",
          "url": "https://scout.docker.com/v/CVE-2021-3449?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1k-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-3449?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1k-r0"
        }
      ]
    },
    {
      "id": "ddd491860dc1580df2b89a2ade973c8ccad80ffa1914a6b7c8b8c5932f0da4fc",
      "category": "container_scanning",
      "message": "CVE-2022-23491 on certifi@2020.4.5.1",
      "description": "Certifi 2022.12.07 removes root certificates from \"TrustCor\" from the root store. These are in the process of being removed from Mozilla's trust store.\n\nTrustCor's root certificates are being removed pursuant to an investigation prompted by media reporting that TrustCor's ownership also operated a business that produced spyware. Conclusions of Mozilla's investigation can be found [here](https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/oxX69KFvsm4/m/yLohoVqtCgAJ).",
      "cve": "CVE-2022-23491",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade certifi@2020.4.5.1 to 2022.12.07",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/certifi@2020.4.5.1"
          },
          "version": "2020.4.5.1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2022-23491",
          "value": "CVE-2022-23491",
          "url": "https://scout.docker.com/v/CVE-2022-23491?s=github&n=certifi&t=pypi&vr=%3E%3D2017.11.05%2C%3C2022.12.07"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2022-23491?s=github&n=certifi&t=pypi&vr=%3E%3D2017.11.05%2C%3C2022.12.07"
        }
      ]
    },
    {
      "id": "2a4df6243a4a5afe9c9a4ee46a611ddb544e37313a75f9ba886ca5301b5e54e8",
      "category": "container_scanning",
      "message": "CVE-2025-8869 on pip@20.0.2",
      "description": "### Summary\n\nIn the fallback extraction path for source distributions, `pip` used Python’s `tarfile` module without verifying that symbolic/hard link targets resolve inside the intended extraction directory. A malicious sdist can include links that escape the target directory and overwrite arbitrary files on the invoking host during `pip install`.\n\n### Impact\n\nSuccessful exploitation enables arbitrary file overwrite outside the build/extraction directory on the machine running `pip`. This can be leveraged to tamper with configuration or startup files and may lead to further code execution depending on the environment, but the direct, guaranteed impact is integrity compromise on the vulnerable system.\n\n### Conditions\n\nThe issue is triggered when installing an attacker-controlled sdist (e.g., from an index or URL) and the fallback extraction code path is used. No special privileges are required beyond running `pip install`; active user action is necessary.\n\n### Remediation\n\nThe [fix](https://github.com/pypa/pip/pull/13550), while available as a patch that can be manually applied, has not yet been put into a numbered version but is planned for `25.3`. Using a Python interpreter that implements the safe-extraction behavior described by **PEP 706** provides additional defense in depth for other `tarfile` issues but is not a substitute for upgrading pip for this specific flaw.",
      "cve": "CVE-2025-8869",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/pip@20.0.2"
          },
          "version": "20.0.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2025-8869",
          "value": "CVE-2025-8869",
          "url": "https://scout.docker.com/v/CVE-2025-8869?s=github&n=pip&t=pypi&vr=%3C%3D25.2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2025-8869?s=github&n=pip&t=pypi&vr=%3C%3D25.2"
        }
      ]
    },
    {
      "id": "3f1d12a475ba5c0af52d79d340d6ebc139d6b652184710349605cafff957e904",
      "category": "container_scanning",
      "message": "CVE-2020-35494 on alpine/binutils@2.33.1-r0",
      "cve": "CVE-2020-35494",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/binutils@2.33.1-r0?os_name=alpine&os_version=3.11"
          },
          "version": "2.33.1-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-35494",
          "value": "CVE-2020-35494",
          "url": "https://scout.docker.com/v/CVE-2020-35494?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-35494?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ]
    },
    {
      "id": "7bb091b233c87a70ecb073600283c996107ea59b44341b4e6c3eeaec0ea2d7e0",
      "category": "container_scanning",
      "message": "CVE-2023-32681 on requests@2.23.0",
      "description": "### Impact\n\nSince Requests v2.3.0, Requests has been vulnerable to potentially leaking `Proxy-Authorization` headers to destination servers, specifically during redirects to an HTTPS origin. This is a product of how `rebuild_proxies` is used to recompute and [reattach the `Proxy-Authorization` header](https://github.com/psf/requests/blob/f2629e9e3c7ce3c3c8c025bcd8db551101cbc773/requests/sessions.py#L319-L328) to requests when redirected. Note this behavior has _only_ been observed to affect proxied requests when credentials are supplied in the URL user information component (e.g. `https://username:password@proxy:8080`).\n\n**Current vulnerable behavior(s):**\n\n1. HTTP → HTTPS: **leak**\n2. HTTPS → HTTP: **no leak**\n3. HTTPS → HTTPS: **leak**\n4. HTTP → HTTP: **no leak**\n\nFor HTTP connections sent through the proxy, the proxy will identify the header in the request itself and remove it prior to forwarding to the destination server. However when sent over HTTPS, the `Proxy-Authorization` header must be sent in the CONNECT request as the proxy has no visibility into further tunneled requests. This results in Requests forwarding the header to the destination server unintentionally, allowing a malicious actor to potentially exfiltrate those credentials.\n\nThe reason this currently works for HTTPS connections in Requests is the `Proxy-Authorization` header is also handled by urllib3 with our usage of the ProxyManager in adapters.py with [`proxy_manager_for`](https://github.com/psf/requests/blob/f2629e9e3c7ce3c3c8c025bcd8db551101cbc773/requests/adapters.py#L199-L235). This will compute the required proxy headers in `proxy_headers` and pass them to the Proxy Manager, avoiding attaching them directly to the Request object. This will be our preferred option going forward for default usage.\n\n### Patches\nStarting in Requests v2.31.0, Requests will no longer attach this header to redirects with an HTTPS destination. This should have no negative impacts on the default behavior of the library as the proxy credentials are already properly being handled by urllib3's ProxyManager.\n\nFor users with custom adapters, this _may_ be potentially breaking if you were already working around this behavior. The previous functionality of `rebuild_proxies` doesn't make sense in any case, so we would encourage any users impacted to migrate any handling of Proxy-Authorization directly into their custom adapter.\n\n### Workarounds\nFor users who are not able to update Requests immediately, there is one potential workaround.\n\nYou may disable redirects by setting `allow_redirects` to `False` on all calls through Requests top-level APIs. Note that if you're currently relying on redirect behaviors, you will need to capture the 3xx response codes and ensure a new request is made to the redirect destination.\n```\nimport requests\nr = requests.get('http://github.com/', allow_redirects=False)\n```\n\n### Credits\n\nThis vulnerability was discovered and disclosed by the following individuals.\n\nDennis Brinkrolf, Haxolot (https://haxolot.com/)\nTobias Funke, (tobiasfunke93@gmail.com)",
      "cve": "CVE-2023-32681",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade requests@2.23.0 to 2.31.0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/requests@2.23.0"
          },
          "version": "2.23.0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2023-32681",
          "value": "CVE-2023-32681",
          "url": "https://scout.docker.com/v/CVE-2023-32681?s=github&n=requests&t=pypi&vr=%3E%3D2.3.0%2C%3C2.31.0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2023-32681?s=github&n=requests&t=pypi&vr=%3E%3D2.3.0%2C%3C2.31.0"
        }
      ]
    },
    {
      "id": "3b1f8aa9a8400b53b0199a2fb10e1229415780df724d7ab678ce3ae947f17727",
      "category": "container_scanning",
      "message": "CVE-2021-20197 on alpine/binutils@2.33.1-r0",
      "cve": "CVE-2021-20197",
      "severity": "Medium",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/binutils@2.33.1-r0?os_name=alpine&os_version=3.11"
          },
          "version": "2.33.1-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-20197",
          "value": "CVE-2021-20197",
          "url": "https://scout.docker.com/v/CVE-2021-20197?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-20197?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ]
    },
    {
      "id": "5cce251ba6db7b89fd764bd64124ff9c85c641a13f3bfde08e553ffd2349e630",
      "category": "container_scanning",
      "message": "GHSA-w235-7p84-xx57 on tornado@4.5.3",
      "description": "### Summary\nTornado’s `curl_httpclient.CurlAsyncHTTPClient` class is vulnerable to CRLF (carriage return/line feed) injection in the request headers.\n\n### Details\nWhen an HTTP request is sent using `CurlAsyncHTTPClient`, Tornado does not reject carriage return (\\r) or line feed (\\n) characters in the request headers. As a result, if an application includes an attacker-controlled header value in a request sent using `CurlAsyncHTTPClient`, the attacker can inject arbitrary headers into the request or cause the application to send arbitrary requests to the specified server.\n\nThis behavior differs from that of the standard `AsyncHTTPClient` class, which does reject CRLF characters.\n\nThis issue appears to stem from libcurl's (as well as pycurl's) lack of validation for the [`HTTPHEADER`](https://curl.se/libcurl/c/CURLOPT_HTTPHEADER.html) option. libcurl’s documentation states:\n\n> The headers included in the linked list must not be CRLF-terminated, because libcurl adds CRLF after each header item itself. Failure to comply with this might result in strange behavior. libcurl passes on the verbatim strings you give it, without any filter or other safe guards. That includes white space and control characters.\n\npycurl similarly appears to assume that the headers adhere to the correct format. Therefore, without any validation on Tornado’s part, header names and values are included verbatim in the request sent by `CurlAsyncHTTPClient`, including any control characters that have special meaning in HTTP semantics.\n\n### PoC\nThe issue can be reproduced using the following script:\n\n```python\nimport asyncio\n\nfrom tornado import httpclient\nfrom tornado import curl_httpclient\n\nasync def main():\n    http_client = curl_httpclient.CurlAsyncHTTPClient()\n\n    request = httpclient.HTTPRequest(\n        # Burp Collaborator payload\n        \"http://727ymeu841qydmnwlol261ktkkqbe24qt.oastify.com/\",\n        method=\"POST\",\n        body=\"body\",\n        # Injected header using CRLF characters\n        headers={\"Foo\": \"Bar\\r\\nHeader: Injected\"}\n    )\n\n    response = await http_client.fetch(request)\n    print(response.body)\n\n    http_client.close()\n\nif __name__ == \"__main__\":\n    asyncio.run(main())\n```\n\nWhen the specified server receives the request, it contains the injected header (`Header: Injected`) on its own line:\n\n```http\nPOST / HTTP/1.1\nHost: 727ymeu841qydmnwlol261ktkkqbe24qt.oastify.com\nUser-Agent: Mozilla/5.0 (compatible; pycurl)\nAccept: */*\nAccept-Encoding: gzip,deflate\nFoo: Bar\nHeader: Injected\nContent-Length: 4\nContent-Type: application/x-www-form-urlencoded\n\nbody\n```\n\nThe attacker can also construct entirely new requests using a payload with multiple CRLF sequences. For example, specifying a header value of `\\r\\n\\r\\nPOST /attacker-controlled-url HTTP/1.1\\r\\nHost: 727ymeu841qydmnwlol261ktkkqbe24qt.oastify.com` results in the server receiving an additional, attacker-controlled request:\n\n```http\nPOST /attacker-controlled-url HTTP/1.1\nHost: 727ymeu841qydmnwlol261ktkkqbe24qt.oastify.com\nContent-Length: 4\nContent-Type: application/x-www-form-urlencoded\n\nbody\n```\n\n### Impact\nApplications using the Tornado library to send HTTP requests with untrusted header data are affected. This issue may facilitate the exploitation of server-side request forgery (SSRF) vulnerabilities.",
      "cve": "GHSA-w235-7p84-xx57",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade tornado@4.5.3 to 6.4.1",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/tornado@4.5.3"
          },
          "version": "4.5.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "GHSA-w235-7p84-xx57",
          "value": "GHSA-w235-7p84-xx57",
          "url": "https://scout.docker.com/v/GHSA-w235-7p84-xx57?s=github&n=tornado&t=pypi&vr=%3C%3D6.4.0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/GHSA-w235-7p84-xx57?s=github&n=tornado&t=pypi&vr=%3C%3D6.4.0"
        }
      ]
    },
    {
      "id": "f81e55800d963d914802f0dde690bb5a1830e62971e798603b8bcba73187ce58",
      "category": "container_scanning",
      "message": "CVE-2023-34049 on salt@2019.2.3",
      "description": "The Salt-SSH pre-flight option copies the script to the target at a predictable path, which allows an attacker to force Salt-SSH to run their script. If an attacker has access to the target VM and knows the path to the pre-flight script before it runs they can ensure Salt-SSH runs their script with the privileges of the user running Salt-SSH. Do not make the copy path on the target predictable and ensure we check return codes of the scp command if the copy fails.",
      "cve": "CVE-2023-34049",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3005.4",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2023-34049",
          "value": "CVE-2023-34049",
          "url": "https://scout.docker.com/v/CVE-2023-34049?s=github&n=salt&t=pypi&vr=%3C3005.4"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2023-34049?s=github&n=salt&t=pypi&vr=%3C3005.4"
        }
      ]
    },
    {
      "id": "de9d7d56f3dfd56282f742ee9c86cb3e847b7e6083921950384d84545d0ae914",
      "category": "container_scanning",
      "message": "CVE-2023-5752 on pip@20.0.2",
      "description": "When installing a package from a Mercurial VCS URL, e.g. `pip install hg+...`, with pip prior to v23.3, the specified Mercurial revision could be used to inject arbitrary configuration options to the `hg clone` call (e.g. `--config`). Controlling the Mercurial configuration can modify how and which repository is installed. This vulnerability does not affect users who aren't installing from Mercurial.",
      "cve": "CVE-2023-5752",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade pip@20.0.2 to 23.3",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/pip@20.0.2"
          },
          "version": "20.0.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2023-5752",
          "value": "CVE-2023-5752",
          "url": "https://scout.docker.com/v/CVE-2023-5752?s=github&n=pip&t=pypi&vr=%3C23.3"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2023-5752?s=github&n=pip&t=pypi&vr=%3C23.3"
        }
      ]
    },
    {
      "id": "181aa29186e4614d61529dfdb8c68044919fc65cdee333eeb2926d0dc5a86ca5",
      "category": "container_scanning",
      "message": "CVE-2020-28493 on jinja2@2.11.2",
      "description": "This affects the package jinja2 from 0.0.0 and before 2.11.3. The ReDOS vulnerability of the regex is mainly due to the sub-pattern [a-zA-Z0-9._-]+.[a-zA-Z0-9._-]+ This issue can be mitigated by Markdown to format user content instead of the urlize filter, or by implementing request timeouts and limiting process memory.",
      "cve": "CVE-2020-28493",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade jinja2@2.11.2 to 2.11.3",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/jinja2@2.11.2"
          },
          "version": "2.11.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-28493",
          "value": "CVE-2020-28493",
          "url": "https://scout.docker.com/v/CVE-2020-28493?s=github&n=jinja2&t=pypi&vr=%3C2.11.3"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-28493?s=github&n=jinja2&t=pypi&vr=%3C2.11.3"
        }
      ]
    },
    {
      "id": "baae98bf9e226cd58d323d6a808a1989752b41cac01eae81aab17184bc822e6a",
      "category": "container_scanning",
      "message": "CVE-2023-23931 on cryptography@2.9.2",
      "description": "Previously, `Cipher.update_into` would accept Python objects which implement the buffer protocol, but provide only immutable buffers:\n\n```pycon\n>>> outbuf = b\"\\x00\" * 32\n>>> c = ciphers.Cipher(AES(b\"\\x00\" * 32), modes.ECB()).encryptor()\n>>> c.update_into(b\"\\x00\" * 16, outbuf)\n16\n>>> outbuf\nb'\\xdc\\x95\\xc0x\\xa2@\\x89\\x89\\xadH\\xa2\\x14\\x92\\x84 \\x87\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00'\n```\n\nThis would allow immutable objects (such as `bytes`) to be mutated, thus violating fundamental rules of Python. This is a soundness bug -- it allows programmers to misuse an API, it cannot be exploited by attacker controlled data alone.\n\nThis now correctly raises an exception.\n\nThis issue has been present since `update_into` was originally introduced in cryptography 1.8.",
      "cve": "CVE-2023-23931",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade cryptography@2.9.2 to 39.0.1",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/cryptography@2.9.2"
          },
          "version": "2.9.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2023-23931",
          "value": "CVE-2023-23931",
          "url": "https://scout.docker.com/v/CVE-2023-23931?s=github&n=cryptography&t=pypi&vr=%3E%3D1.8%2C%3C39.0.1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2023-23931?s=github&n=cryptography&t=pypi&vr=%3E%3D1.8%2C%3C39.0.1"
        }
      ]
    },
    {
      "id": "4209d5618cd3e4c938b59eb90e713283ac79a57e9486cdf2d09f9684aba5129c",
      "category": "container_scanning",
      "message": "CVE-2024-3651 on idna@2.9",
      "description": "### Impact\nA specially crafted argument to the `idna.encode()` function could consume significant resources. This may lead to a denial-of-service.\n\n### Patches\nThe function has been refined to reject such strings without the associated resource consumption in version 3.7.\n\n### Workarounds\nDomain names cannot exceed 253 characters in length, if this length limit is enforced prior to passing the domain to the `idna.encode()` function it should no longer consume significant resources. This is triggered by arbitrarily large inputs that would not occur in normal usage, but may be passed to the library assuming there is no preliminary input validation by the higher-level application.\n\n### References\n* https://huntr.com/bounties/93d78d07-d791-4b39-a845-cbfabc44aadb",
      "cve": "CVE-2024-3651",
      "severity": "Medium",
      "confidence": "Unknown",
      "solution": "Upgrade idna@2.9 to 3.7",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/idna@2.9"
          },
          "version": "2.9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2024-3651",
          "value": "CVE-2024-3651",
          "url": "https://scout.docker.com/v/CVE-2024-3651?s=github&n=idna&t=pypi&vr=%3C3.7"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2024-3651?s=github&n=idna&t=pypi&vr=%3C3.7"
        }
      ]
    },
    {
      "id": "c39db4bce16935a64e97e77c553b9b84e9d26adb406d034edfb5183898810c4b",
      "category": "container_scanning",
      "message": "CVE-2020-13630 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2020-13630",
      "severity": "High",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-13630",
          "value": "CVE-2020-13630",
          "url": "https://scout.docker.com/v/CVE-2020-13630?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-13630?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "8ab896111aceb2a1cbe1bde61bd2401c8872fae9e131bd79f1111a7921aa6fd8",
      "category": "container_scanning",
      "message": "CVE-2020-11652 on salt@2019.2.3",
      "description": "An issue was discovered in SaltStack Salt before 2019.2.4 and 3000 before 3000.2. The salt-master process ClearFuncs class allows access to some methods that improperly sanitize paths. These methods allow arbitrary directory access to authenticated users.",
      "cve": "CVE-2020-11652",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.4",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-11652",
          "value": "CVE-2020-11652",
          "url": "https://scout.docker.com/v/CVE-2020-11652?s=github&n=salt&t=pypi&vr=%3C2019.2.4"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-11652?s=github&n=salt&t=pypi&vr=%3C2019.2.4"
        }
      ]
    },
    {
      "id": "baf38e49ab28e4ecd430c276ebb242b026e8e8e27306dc8add97b5256881b69c",
      "category": "container_scanning",
      "message": "CVE-2021-3572 on pip@20.0.2",
      "description": "A flaw was found in python-pip in the way it handled Unicode separators in git references. A remote attacker could possibly use this issue to install a different revision on a repository. The highest threat from this vulnerability is to data integrity. This is fixed in python-pip version 21.1.",
      "cve": "CVE-2021-3572",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade pip@20.0.2 to 21.1",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/pip@20.0.2"
          },
          "version": "20.0.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-3572",
          "value": "CVE-2021-3572",
          "url": "https://scout.docker.com/v/CVE-2021-3572?s=github&n=pip&t=pypi&vr=%3C21.1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-3572?s=github&n=pip&t=pypi&vr=%3C21.1"
        }
      ]
    },
    {
      "id": "eb79e7121cb5aad9d0c6d2bdf70d7d69c1378f78501f099a67caa638e75dba78",
      "category": "container_scanning",
      "message": "CVE-2023-52323 on pycryptodomex@3.9.7",
      "description": "PyCryptodome and pycryptodomex before 3.19.1 allow side-channel leakage for OAEP decryption, exploitable for a Manger attack.",
      "cve": "CVE-2023-52323",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade pycryptodomex@3.9.7 to 3.19.1",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/pycryptodomex@3.9.7"
          },
          "version": "3.9.7"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2023-52323",
          "value": "CVE-2023-52323",
          "url": "https://scout.docker.com/v/CVE-2023-52323?s=github&n=pycryptodomex&t=pypi&vr=%3C3.19.1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2023-52323?s=github&n=pycryptodomex&t=pypi&vr=%3C3.19.1"
        }
      ]
    },
    {
      "id": "f1c102fe1c754268e6b3c74e57d9f65150a921aeb0a9bc0d3670afba9d896537",
      "category": "container_scanning",
      "message": "CVE-2021-42378 on alpine/busybox@1.31.1-r9",
      "cve": "CVE-2021-42378",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/busybox@1.31.1-r9 to 1.31.1-r11",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/busybox@1.31.1-r9?os_name=alpine&os_version=3.11"
          },
          "version": "1.31.1-r9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-42378",
          "value": "CVE-2021-42378",
          "url": "https://scout.docker.com/v/CVE-2021-42378?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-42378?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ]
    },
    {
      "id": "0348b1e697afb38e8afb3c32af41e778bea6ffb892a1c8adfd5c41daa13219e6",
      "category": "container_scanning",
      "message": "CVE-2021-42379 on alpine/busybox@1.31.1-r9",
      "cve": "CVE-2021-42379",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/busybox@1.31.1-r9 to 1.31.1-r11",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/busybox@1.31.1-r9?os_name=alpine&os_version=3.11"
          },
          "version": "1.31.1-r9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-42379",
          "value": "CVE-2021-42379",
          "url": "https://scout.docker.com/v/CVE-2021-42379?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-42379?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ]
    },
    {
      "id": "96b54cdae355d5faaffef8892386b8edce3e4a9c5f7967c0c96e02ca6fcb1e54",
      "category": "container_scanning",
      "message": "CVE-2021-42380 on alpine/busybox@1.31.1-r9",
      "cve": "CVE-2021-42380",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/busybox@1.31.1-r9 to 1.31.1-r11",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/busybox@1.31.1-r9?os_name=alpine&os_version=3.11"
          },
          "version": "1.31.1-r9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-42380",
          "value": "CVE-2021-42380",
          "url": "https://scout.docker.com/v/CVE-2021-42380?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-42380?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ]
    },
    {
      "id": "4db7e08a01fa15142239c1da1082e84f819dfa51e2b808acf508a81a6e00f0ac",
      "category": "container_scanning",
      "message": "CVE-2021-42381 on alpine/busybox@1.31.1-r9",
      "cve": "CVE-2021-42381",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/busybox@1.31.1-r9 to 1.31.1-r11",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/busybox@1.31.1-r9?os_name=alpine&os_version=3.11"
          },
          "version": "1.31.1-r9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-42381",
          "value": "CVE-2021-42381",
          "url": "https://scout.docker.com/v/CVE-2021-42381?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-42381?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ]
    },
    {
      "id": "a1551a462cfaaa4d4ce7a0338d93b164a803d85256df13e4b9a9d1d58dc32327",
      "category": "container_scanning",
      "message": "CVE-2021-42382 on alpine/busybox@1.31.1-r9",
      "cve": "CVE-2021-42382",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/busybox@1.31.1-r9 to 1.31.1-r11",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/busybox@1.31.1-r9?os_name=alpine&os_version=3.11"
          },
          "version": "1.31.1-r9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-42382",
          "value": "CVE-2021-42382",
          "url": "https://scout.docker.com/v/CVE-2021-42382?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-42382?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ]
    },
    {
      "id": "cec61442d392238fb1bf28eff421908bbfa233f271a6a7ef87baa48996036d74",
      "category": "container_scanning",
      "message": "CVE-2021-42383 on alpine/busybox@1.31.1-r9",
      "cve": "CVE-2021-42383",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/busybox@1.31.1-r9 to 1.31.1-r11",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/busybox@1.31.1-r9?os_name=alpine&os_version=3.11"
          },
          "version": "1.31.1-r9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-42383",
          "value": "CVE-2021-42383",
          "url": "https://scout.docker.com/v/CVE-2021-42383?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-42383?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ]
    },
    {
      "id": "9f0c71080f81c241334d49e26cbd31348dffadc30d85b99601a8ce65408c920a",
      "category": "container_scanning",
      "message": "CVE-2021-42384 on alpine/busybox@1.31.1-r9",
      "cve": "CVE-2021-42384",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/busybox@1.31.1-r9 to 1.31.1-r11",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/busybox@1.31.1-r9?os_name=alpine&os_version=3.11"
          },
          "version": "1.31.1-r9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-42384",
          "value": "CVE-2021-42384",
          "url": "https://scout.docker.com/v/CVE-2021-42384?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-42384?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ]
    },
    {
      "id": "04196929e6ed79e99f96ab678f8033c083aac957c8bb36d4edc1131e940d6a7c",
      "category": "container_scanning",
      "message": "CVE-2021-42385 on alpine/busybox@1.31.1-r9",
      "cve": "CVE-2021-42385",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/busybox@1.31.1-r9 to 1.31.1-r11",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/busybox@1.31.1-r9?os_name=alpine&os_version=3.11"
          },
          "version": "1.31.1-r9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-42385",
          "value": "CVE-2021-42385",
          "url": "https://scout.docker.com/v/CVE-2021-42385?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-42385?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ]
    },
    {
      "id": "828490750d5c4c310742d1c9d90a8e7bdae3ef26596a969b4471952982609f9d",
      "category": "container_scanning",
      "message": "CVE-2021-42386 on alpine/busybox@1.31.1-r9",
      "cve": "CVE-2021-42386",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/busybox@1.31.1-r9 to 1.31.1-r11",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/busybox@1.31.1-r9?os_name=alpine&os_version=3.11"
          },
          "version": "1.31.1-r9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-42386",
          "value": "CVE-2021-42386",
          "url": "https://scout.docker.com/v/CVE-2021-42386?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-42386?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r11"
        }
      ]
    },
    {
      "id": "37cac61b67756bd62c17703636121dca6d0afa703dc6212c255c759fd64c21ae",
      "category": "container_scanning",
      "message": "CVE-2020-35662 on salt@2019.2.3",
      "description": "In SaltStack Salt before 3002.5, when authenticating to services using certain modules, the SSL certificate is not always validated.",
      "cve": "CVE-2020-35662",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-35662",
          "value": "CVE-2020-35662",
          "url": "https://scout.docker.com/v/CVE-2020-35662?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-35662?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ]
    },
    {
      "id": "10bf35d349cd4f9b6b6f51f36dda5a7b952b483973faa1ad163c5352ff018326",
      "category": "container_scanning",
      "message": "CVE-2021-3450 on alpine/openssl@1.1.1g-r0",
      "cve": "CVE-2021-3450",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/openssl@1.1.1g-r0 to 1.1.1k-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/openssl@1.1.1g-r0?os_name=alpine&os_version=3.11"
          },
          "version": "1.1.1g-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-3450",
          "value": "CVE-2021-3450",
          "url": "https://scout.docker.com/v/CVE-2021-3450?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1k-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-3450?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1k-r0"
        }
      ]
    },
    {
      "id": "e993f34d0d8b348d65e98fe8a0eb141be18ee924eab8846ca0e4f5d9dc4d1eaa",
      "category": "container_scanning",
      "message": "CVE-2021-3712 on alpine/openssl@1.1.1g-r0",
      "cve": "CVE-2021-3712",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/openssl@1.1.1g-r0 to 1.1.1l-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/openssl@1.1.1g-r0?os_name=alpine&os_version=3.11"
          },
          "version": "1.1.1g-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-3712",
          "value": "CVE-2021-3712",
          "url": "https://scout.docker.com/v/CVE-2021-3712?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1l-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-3712?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1l-r0"
        }
      ]
    },
    {
      "id": "0334e8ae52cdb7bcb05edc48e8e9b9ef88ab2ea6d19d975be73bbb42d02be26e",
      "category": "container_scanning",
      "message": "CVE-2023-0286 on cryptography@2.9.2",
      "description": "pyca/cryptography's wheels include a statically linked copy of OpenSSL. The versions of OpenSSL included in cryptography 0.8.1-39.0.0  are vulnerable to a security issue. More details about the vulnerabilities themselves can be found in https://www.openssl.org/news/secadv/20221213.txt and https://www.openssl.org/news/secadv/20230207.txt.\n\nIf you are building cryptography source (\"sdist\") then you are responsible for upgrading your copy of OpenSSL. Only users installing from wheels built by the cryptography project (i.e., those distributed on PyPI) need to update their cryptography versions.",
      "cve": "CVE-2023-0286",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade cryptography@2.9.2 to 39.0.1",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/cryptography@2.9.2"
          },
          "version": "2.9.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2023-0286",
          "value": "CVE-2023-0286",
          "url": "https://scout.docker.com/v/CVE-2023-0286?s=github&n=cryptography&t=pypi&vr=%3E%3D0.8.1%2C%3C39.0.1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2023-0286?s=github&n=cryptography&t=pypi&vr=%3E%3D0.8.1%2C%3C39.0.1"
        }
      ]
    },
    {
      "id": "5ecbff2a7381f4584ddbcae110a125a269a0380df5d8484fa5a484f986a11f61",
      "category": "container_scanning",
      "message": "CVE-2023-43804 on urllib3@1.25.9",
      "description": "urllib3 doesn't treat the `Cookie` HTTP header special or provide any helpers for managing cookies over HTTP, that is the responsibility of the user. However, it is possible for a user to specify a `Cookie` header and unknowingly leak information via HTTP redirects to a different origin if that user doesn't disable redirects explicitly.\n\nUsers **must** handle redirects themselves instead of relying on urllib3's automatic redirects to achieve safe processing of the `Cookie` header, thus we decided to strip the header by default in order to further protect users who aren't using the correct approach.\n\n## Affected usages\n\nWe believe the number of usages affected by this advisory is low. It requires all of the following to be true to be exploited:\n\n* Using an affected version of urllib3 (patched in v1.26.17 and v2.0.6)\n* Using the `Cookie` header on requests, which is mostly typical for impersonating a browser.\n* Not disabling HTTP redirects\n* Either not using HTTPS or for the origin server to redirect to a malicious origin.\n\n## Remediation\n\n* Upgrading to at least urllib3 v1.26.17 or v2.0.6\n* Disabling HTTP redirects using `redirects=False` when sending requests.\n* Not using the `Cookie` header.",
      "cve": "CVE-2023-43804",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade urllib3@1.25.9 to 1.26.17",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/urllib3@1.25.9"
          },
          "version": "1.25.9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2023-43804",
          "value": "CVE-2023-43804",
          "url": "https://scout.docker.com/v/CVE-2023-43804?s=github&n=urllib3&t=pypi&vr=%3C1.26.17"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2023-43804?s=github&n=urllib3&t=pypi&vr=%3C1.26.17"
        }
      ]
    },
    {
      "id": "60a1ad6e0e80ff20f9e37eaadd662d515edee1736c49928ad201d22e5e779db7",
      "category": "container_scanning",
      "message": "CVE-2019-15847 on alpine/gcc@9.2.0-r4",
      "cve": "CVE-2019-15847",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/gcc@9.2.0-r4 to 9.3.0-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/gcc@9.2.0-r4?os_name=alpine&os_version=3.11"
          },
          "version": "9.2.0-r4"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-15847",
          "value": "CVE-2019-15847",
          "url": "https://scout.docker.com/v/CVE-2019-15847?s=alpine&n=gcc&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C9.3.0-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-15847?s=alpine&n=gcc&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C9.3.0-r0"
        }
      ]
    },
    {
      "id": "23dbf30e16cbe77450673a30fe88c3b5f7c9ea886f4b10d0c38f5b9c920e68a1",
      "category": "container_scanning",
      "message": "CVE-2019-19603 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2019-19603",
      "severity": "High",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-19603",
          "value": "CVE-2019-19603",
          "url": "https://scout.docker.com/v/CVE-2019-19603?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-19603?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "200af07a570e0f4fc01051e6a9b31c4507dceadde9d869976ca94de54ad2c020",
      "category": "container_scanning",
      "message": "CVE-2019-19880 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2019-19880",
      "severity": "High",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-19880",
          "value": "CVE-2019-19880",
          "url": "https://scout.docker.com/v/CVE-2019-19880?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-19880?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "24352ca75adf50077fcb373480bb77be70e8a096acd4381d39970ae865b3b69a",
      "category": "container_scanning",
      "message": "CVE-2019-19923 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2019-19923",
      "severity": "High",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-19923",
          "value": "CVE-2019-19923",
          "url": "https://scout.docker.com/v/CVE-2019-19923?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-19923?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "5b4c8281c42f15e9888183dbd3eae250c1dc16c6556b73db8227dd5ac814b3ff",
      "category": "container_scanning",
      "message": "CVE-2019-19925 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2019-19925",
      "severity": "High",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-19925",
          "value": "CVE-2019-19925",
          "url": "https://scout.docker.com/v/CVE-2019-19925?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-19925?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "64d696bab1c9d7bd31269a0142bd5b0f0b1f4bd765d16b5b8b6490b159c129df",
      "category": "container_scanning",
      "message": "CVE-2019-19926 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2019-19926",
      "severity": "High",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-19926",
          "value": "CVE-2019-19926",
          "url": "https://scout.docker.com/v/CVE-2019-19926?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-19926?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "ba093b7aaf8a013377702ed95db5b7dc4a02ddf360b613289a15e4bc63ca5bc8",
      "category": "container_scanning",
      "message": "CVE-2019-19959 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2019-19959",
      "severity": "High",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-19959",
          "value": "CVE-2019-19959",
          "url": "https://scout.docker.com/v/CVE-2019-19959?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-19959?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "97e947597b4db9e2cdf29857144f2a8d610ad9d74bf1bf0f1b7e444b7495642d",
      "category": "container_scanning",
      "message": "CVE-2019-20218 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2019-20218",
      "severity": "High",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-20218",
          "value": "CVE-2019-20218",
          "url": "https://scout.docker.com/v/CVE-2019-20218?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-20218?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "822fb8b40aeebc808694af3246f55cac52ab4382554dd329a83b5c7ccc7d86da",
      "category": "container_scanning",
      "message": "CVE-2020-11655 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2020-11655",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/sqlite@3.30.1-r1 to 3.30.1-r2",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-11655",
          "value": "CVE-2020-11655",
          "url": "https://scout.docker.com/v/CVE-2020-11655?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-11655?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C3.30.1-r2"
        }
      ]
    },
    {
      "id": "3ac7aa9bcbb2fbea2484659a2cf6b22ae14a41d118d86cb47e5b9069ae006699",
      "category": "container_scanning",
      "message": "CVE-2020-12723 on alpine/perl@5.30.1-r0",
      "cve": "CVE-2020-12723",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/perl@5.30.1-r0 to 5.30.3-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/perl@5.30.1-r0?os_name=alpine&os_version=3.11"
          },
          "version": "5.30.1-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-12723",
          "value": "CVE-2020-12723",
          "url": "https://scout.docker.com/v/CVE-2020-12723?s=alpine&n=perl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C5.30.3-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-12723?s=alpine&n=perl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C5.30.3-r0"
        }
      ]
    },
    {
      "id": "5f624701f6245096faf34095388407498a4803d61cc38141789a6d28c7dad752",
      "category": "container_scanning",
      "message": "CVE-2020-28196 on alpine/krb5@1.17.1-r0",
      "cve": "CVE-2020-28196",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/krb5@1.17.1-r0 to 1.17.2-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/krb5@1.17.1-r0?os_name=alpine&os_version=3.11"
          },
          "version": "1.17.1-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-28196",
          "value": "CVE-2020-28196",
          "url": "https://scout.docker.com/v/CVE-2020-28196?s=alpine&n=krb5&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.17.2-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-28196?s=alpine&n=krb5&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.17.2-r0"
        }
      ]
    },
    {
      "id": "f2bba63ad8950588c5db74f2073b1bca94bffbf215caf39b60cc80044395ff6d",
      "category": "container_scanning",
      "message": "CVE-2021-21996 on salt@2019.2.3",
      "description": "An issue was discovered in SaltStack Salt before 3003.3. A user who has control of the source, and source_hash URLs can gain full file system access as root on a salt minion.",
      "cve": "CVE-2021-21996",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3003.3",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-21996",
          "value": "CVE-2021-21996",
          "url": "https://scout.docker.com/v/CVE-2021-21996?s=github&n=salt&t=pypi&vr=%3C3003.3"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-21996?s=github&n=salt&t=pypi&vr=%3C3003.3"
        }
      ]
    },
    {
      "id": "7f82ee0de29decc9a8aee49ea7817fbf239d20b85c7966fb006bbdbc6ac28ef9",
      "category": "container_scanning",
      "message": "CVE-2021-22004 on salt@2019.2.3",
      "description": "An issue was discovered in SaltStack Salt before 3003.3. The salt minion installer will accept and use a minion config file at C:\\salt\\conf if that file is in place before the installer is run. This allows for a malicious actor to subvert the proper behaviour of the given minion software.",
      "cve": "CVE-2021-22004",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3003.3",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-22004",
          "value": "CVE-2021-22004",
          "url": "https://scout.docker.com/v/CVE-2021-22004?s=github&n=salt&t=pypi&vr=%3C3003.3"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-22004?s=github&n=salt&t=pypi&vr=%3C3003.3"
        }
      ]
    },
    {
      "id": "a506a401dc6193bfa2c12f9bb8d8217ca3b198d586ca9a3478e624fb067317f9",
      "category": "container_scanning",
      "message": "CVE-2021-23840 on alpine/openssl@1.1.1g-r0",
      "cve": "CVE-2021-23840",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/openssl@1.1.1g-r0 to 1.1.1j-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/openssl@1.1.1g-r0?os_name=alpine&os_version=3.11"
          },
          "version": "1.1.1g-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-23840",
          "value": "CVE-2021-23840",
          "url": "https://scout.docker.com/v/CVE-2021-23840?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1j-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-23840?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1j-r0"
        }
      ]
    },
    {
      "id": "5e713b59a50e504c10a212621e84044188262051cfaa0a779b8fa9f9e8f50f0c",
      "category": "container_scanning",
      "message": "CVE-2021-28831 on alpine/busybox@1.31.1-r9",
      "cve": "CVE-2021-28831",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/busybox@1.31.1-r9 to 1.31.1-r10",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/busybox@1.31.1-r9?os_name=alpine&os_version=3.11"
          },
          "version": "1.31.1-r9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-28831",
          "value": "CVE-2021-28831",
          "url": "https://scout.docker.com/v/CVE-2021-28831?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r10"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-28831?s=alpine&n=busybox&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.31.1-r10"
        }
      ]
    },
    {
      "id": "b72de0e6e4cbdab334a65ccef105d371c7ed8092dc008721e527395283d3ca9a",
      "category": "container_scanning",
      "message": "CVE-2021-30139 on alpine/apk-tools@2.10.5-r0",
      "cve": "CVE-2021-30139",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/apk-tools@2.10.5-r0 to 2.10.6-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/apk-tools@2.10.5-r0?os_name=alpine&os_version=3.11"
          },
          "version": "2.10.5-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-30139",
          "value": "CVE-2021-30139",
          "url": "https://scout.docker.com/v/CVE-2021-30139?s=alpine&n=apk-tools&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C2.10.6-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-30139?s=alpine&n=apk-tools&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C2.10.6-r0"
        }
      ]
    },
    {
      "id": "55be2d286eb2ebad535dca614f74a181a3e923b0b1177353696fbdf2e8ae921d",
      "category": "container_scanning",
      "message": "CVE-2021-43618 on alpine/gmp@6.1.2-r1",
      "cve": "CVE-2021-43618",
      "severity": "High",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/gmp@6.1.2-r1?os_name=alpine&os_version=3.11"
          },
          "version": "6.1.2-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-43618",
          "value": "CVE-2021-43618",
          "url": "https://scout.docker.com/v/CVE-2021-43618?s=alpine&n=gmp&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D6.1.2-r1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-43618?s=alpine&n=gmp&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D6.1.2-r1"
        }
      ]
    },
    {
      "id": "a54ae569361de534f627687c47e7da0a1aacbef147dc2bbf07cafdaf22d4e6ac",
      "category": "container_scanning",
      "message": "CVE-2022-40898 on wheel@0.34.2",
      "description": "Python Packaging Authority (PyPA) Wheel is a reference implementation of the Python wheel packaging standard. Wheel 0.37.1 and earlier are vulnerable to a Regular Expression denial of service via attacker controlled input to the wheel cli. The vulnerable regex is used to verify the validity of Wheel file names. This has been patched in version 0.38.1.",
      "cve": "CVE-2022-40898",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade wheel@0.34.2 to 0.38.1",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/wheel@0.34.2"
          },
          "version": "0.34.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2022-40898",
          "value": "CVE-2022-40898",
          "url": "https://scout.docker.com/v/CVE-2022-40898?s=github&n=wheel&t=pypi&vr=%3C0.38.1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2022-40898?s=github&n=wheel&t=pypi&vr=%3C0.38.1"
        }
      ]
    },
    {
      "id": "5b6b6527a13e8856d4b49506a14ef8f26a2d404d14022f18da82a9c99ebb7e95",
      "category": "container_scanning",
      "message": "CVE-2023-37920 on certifi@2020.4.5.1",
      "description": "Certifi 2023.07.22 removes root certificates from \"e-Tugra\" from the root store. These are in the process of being removed from Mozilla's trust store.\n\n e-Tugra's root certificates are being removed pursuant to an investigation prompted by reporting of security issues in their systems. Conclusions of Mozilla's investigation can be found [here](https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/C-HrP1SEq1A).",
      "cve": "CVE-2023-37920",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade certifi@2020.4.5.1 to 2023.7.22",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/certifi@2020.4.5.1"
          },
          "version": "2020.4.5.1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2023-37920",
          "value": "CVE-2023-37920",
          "url": "https://scout.docker.com/v/CVE-2023-37920?s=github&n=certifi&t=pypi&vr=%3E%3D2015.4.28%2C%3C2023.7.22"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2023-37920?s=github&n=certifi&t=pypi&vr=%3E%3D2015.4.28%2C%3C2023.7.22"
        }
      ]
    },
    {
      "id": "8a471a4a61390d1bf30d1d67b3ee03e1d4d6bae3063ad6777549c3ae2640a83d",
      "category": "container_scanning",
      "message": "CVE-2024-52804 on tornado@4.5.3",
      "description": "The algorithm used for parsing HTTP cookies in Tornado versions prior to 6.4.2 sometimes has quadratic complexity, leading to excessive CPU consumption when parsing maliciously-crafted cookie headers. This parsing occurs in the event loop thread and may block the processing of other requests.\n\nSee also CVE-2024-7592 for a similar vulnerability in cpython.",
      "cve": "CVE-2024-52804",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade tornado@4.5.3 to 6.4.2",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/tornado@4.5.3"
          },
          "version": "4.5.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2024-52804",
          "value": "CVE-2024-52804",
          "url": "https://scout.docker.com/v/CVE-2024-52804?s=github&n=tornado&t=pypi&vr=%3C%3D6.4.1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2024-52804?s=github&n=tornado&t=pypi&vr=%3C%3D6.4.1"
        }
      ]
    },
    {
      "id": "e89f2e12aba3c61823f8697f868e2c50835c45412ce7399e3e9d2454f3b68873",
      "category": "container_scanning",
      "message": "CVE-2024-6345 on setuptools@46.1.3",
      "description": "A vulnerability in the `package_index` module of pypa/setuptools versions up to 69.1.1 allows for remote code execution via its download functions. These functions, which are used to download packages from URLs provided by users or retrieved from package index servers, are susceptible to code injection. If these functions are exposed to user-controlled inputs, such as package URLs, they can execute arbitrary commands on the system. The issue is fixed in version 70.0.",
      "cve": "CVE-2024-6345",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade setuptools@46.1.3 to 70.0.0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/setuptools@46.1.3"
          },
          "version": "46.1.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2024-6345",
          "value": "CVE-2024-6345",
          "url": "https://scout.docker.com/v/CVE-2024-6345?s=github&n=setuptools&t=pypi&vr=%3C70.0.0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2024-6345?s=github&n=setuptools&t=pypi&vr=%3C70.0.0"
        }
      ]
    },
    {
      "id": "df75a03c469019e09189df9aec85f09f556b58c2b39cf18e7bb0f516c95f3233",
      "category": "container_scanning",
      "message": "CVE-2025-47287 on tornado@4.5.3",
      "description": "### Summary\n\nWhen Tornado's ``multipart/form-data`` parser encounters certain errors, it logs a warning but continues trying to parse the remainder of the data. This allows remote attackers to generate an extremely high volume of logs, constituting a DoS attack. This DoS is compounded by the fact that the logging subsystem is synchronous.\n\n### Affected versions\n\nAll versions of Tornado prior to 6.5 are affected. The vulnerable parser is enabled by default.\n\n### Solution\n\nUpgrade to Tornado version 6.5. In the meantime, risk can be mitigated by blocking `Content-Type: multipart/form-data` in a proxy.",
      "cve": "CVE-2025-47287",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade tornado@4.5.3 to 6.5.0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/tornado@4.5.3"
          },
          "version": "4.5.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2025-47287",
          "value": "CVE-2025-47287",
          "url": "https://scout.docker.com/v/CVE-2025-47287?s=github&n=tornado&t=pypi&vr=%3C6.5"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2025-47287?s=github&n=tornado&t=pypi&vr=%3C6.5"
        }
      ]
    },
    {
      "id": "5764aae90af401379b0584099f0360d8e33ddf59dea7a95d18c4c9e3788e6fc7",
      "category": "container_scanning",
      "message": "CVE-2022-22967 on salt@2019.2.3",
      "description": "An issue was discovered in SaltStack Salt in versions before 3002.9, 3003.5, 3004.2. PAM auth fails to reject locked accounts, which allows a previously authorized user whose account is locked still run Salt commands when their account is locked. This affects both local shell accounts with an active session and salt-api users that authenticate via PAM eauth.",
      "cve": "CVE-2022-22967",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3002.9",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2022-22967",
          "value": "CVE-2022-22967",
          "url": "https://scout.docker.com/v/CVE-2022-22967?s=github&n=salt&t=pypi&vr=%3C3002.9"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2022-22967?s=github&n=salt&t=pypi&vr=%3C3002.9"
        }
      ]
    },
    {
      "id": "162c6767d6d5156a36b3b66c4889fde2f5eba34d4609f2c7e8da8b0bcff72730",
      "category": "container_scanning",
      "message": "CVE-2024-22232 on salt@2019.2.3",
      "description": "A specially crafted url can be created which leads to a directory traversal in the salt file server.\nA malicious user can read an arbitrary file from a Salt master’s filesystem.",
      "cve": "CVE-2024-22232",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3005.5",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2024-22232",
          "value": "CVE-2024-22232",
          "url": "https://scout.docker.com/v/CVE-2024-22232?s=github&n=salt&t=pypi&vr=%3C3005.5"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2024-22232?s=github&n=salt&t=pypi&vr=%3C3005.5"
        }
      ]
    },
    {
      "id": "178ef9a76dddfb0347f3353fa742acb657da42ae9ca76a50eab2eef81e5612b6",
      "category": "container_scanning",
      "message": "CVE-2025-47273 on setuptools@46.1.3",
      "description": "### Summary \nA path traversal vulnerability in `PackageIndex` was fixed in setuptools version 78.1.1\n\n### Details\n```\n    def _download_url(self, url, tmpdir):\n        # Determine download filename\n        #\n        name, _fragment = egg_info_for_url(url)\n        if name:\n            while '..' in name:\n                name = name.replace('..', '.').replace('\\\\', '_')\n        else:\n            name = \"__downloaded__\"  # default if URL has no path contents\n\n        if name.endswith('.[egg.zip](http://egg.zip/)'):\n            name = name[:-4]  # strip the extra .zip before download\n\n -->       filename = os.path.join(tmpdir, name)\n```\n\nHere: https://github.com/pypa/setuptools/blob/6ead555c5fb29bc57fe6105b1bffc163f56fd558/setuptools/package_index.py#L810C1-L825C88\n\n`os.path.join()` discards the first argument `tmpdir` if the second begins with a slash or drive letter.\n`name` is derived from a URL without sufficient sanitization. While there is some attempt to sanitize by replacing instances of '..' with '.', it is insufficient.\n\n### Risk Assessment\nAs easy_install and package_index are deprecated, the exploitation surface is reduced.\nHowever, it seems this could be exploited in a similar fashion like https://github.com/advisories/GHSA-r9hx-vwmv-q579, and as described by POC 4 in https://github.com/advisories/GHSA-cx63-2mw6-8hw5 report: via malicious URLs present on the pages of a package index.\n\n### Impact\nAn attacker would be allowed to write files to arbitrary locations on the filesystem with the permissions of the process running the Python code, which could escalate to RCE depending on the context.\n\n### References\nhttps://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5\nhttps://github.com/pypa/setuptools/issues/4946",
      "cve": "CVE-2025-47273",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade setuptools@46.1.3 to 78.1.1",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/setuptools@46.1.3"
          },
          "version": "46.1.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2025-47273",
          "value": "CVE-2025-47273",
          "url": "https://scout.docker.com/v/CVE-2025-47273?s=github&n=setuptools&t=pypi&vr=%3C78.1.1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2025-47273?s=github&n=setuptools&t=pypi&vr=%3C78.1.1"
        }
      ]
    },
    {
      "id": "33a909b936328780993c5d2d539650c0389f94aab120b2005179e3c85720f141",
      "category": "container_scanning",
      "message": "CVE-2020-28243 on salt@2019.2.3",
      "description": "An issue was discovered in SaltStack Salt before 3002.5. The minion's `restartcheck` is vulnerable to command injection via a crafted process name. This allows for a local privilege escalation by any user able to create a files on the minion in a non-blacklisted directory.",
      "cve": "CVE-2020-28243",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-28243",
          "value": "CVE-2020-28243",
          "url": "https://scout.docker.com/v/CVE-2020-28243?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-28243?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ]
    },
    {
      "id": "66bc6d704b346261f1c8e422e905b07772b4634be44e5a4db3d4bb56e7bac209",
      "category": "container_scanning",
      "message": "CVE-2021-20294 on alpine/binutils@2.33.1-r0",
      "cve": "CVE-2021-20294",
      "severity": "High",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/binutils@2.33.1-r0?os_name=alpine&os_version=3.11"
          },
          "version": "2.33.1-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-20294",
          "value": "CVE-2021-20294",
          "url": "https://scout.docker.com/v/CVE-2021-20294?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-20294?s=alpine&n=binutils&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D2.33.1-r1"
        }
      ]
    },
    {
      "id": "1479afc69a7148b3aab36a7cf96721a0b15b66c1038b843a5eeece1c24b86a45",
      "category": "container_scanning",
      "message": "CVE-2021-25315 on salt@2019.2.3",
      "description": "A Incorrect Implementation of Authentication Algorithm vulnerability in of SUSE SUSE Linux Enterprise Server 15 SP 3; openSUSE Tumbleweed allows local attackers to execute arbitrary code via salt without the need to specify valid credentials. This issue affects: SUSE SUSE Linux Enterprise Server 15 SP 3 salt versions prior to 3002.2-3. openSUSE Tumbleweed salt version 3002.2-2.1 and prior versions.",
      "cve": "CVE-2021-25315",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3002.2",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-25315",
          "value": "CVE-2021-25315",
          "url": "https://scout.docker.com/v/CVE-2021-25315?s=github&n=salt&t=pypi&vr=%3C3002.2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-25315?s=github&n=salt&t=pypi&vr=%3C3002.2"
        }
      ]
    },
    {
      "id": "ae310ebba6abb63976179016a5e9bd2135212b3b7d62e55bda609b191d555d92",
      "category": "container_scanning",
      "message": "CVE-2021-31607 on salt@2019.2.3",
      "description": "In SaltStack Salt 2016.9 through 3002.6, a command injection vulnerability exists in the snapper module that allows for local privilege escalation on a minion. The attack requires that a file is created with a pathname that is backed up by snapper, and that the master calls the snapper.diff function (which executes popen unsafely).",
      "cve": "CVE-2021-31607",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3003rc1",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-31607",
          "value": "CVE-2021-31607",
          "url": "https://scout.docker.com/v/CVE-2021-31607?s=github&n=salt&t=pypi&vr=%3E%3D2016.11.0%2C%3C%3D3002.6"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-31607?s=github&n=salt&t=pypi&vr=%3E%3D2016.11.0%2C%3C%3D3002.6"
        }
      ]
    },
    {
      "id": "c784384205311654987dadb2af29c487861b09f22b3a85e8564ba35a0c7793ba",
      "category": "container_scanning",
      "message": "CVE-2020-10543 on alpine/perl@5.30.1-r0",
      "cve": "CVE-2020-10543",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/perl@5.30.1-r0 to 5.30.3-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/perl@5.30.1-r0?os_name=alpine&os_version=3.11"
          },
          "version": "5.30.1-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-10543",
          "value": "CVE-2020-10543",
          "url": "https://scout.docker.com/v/CVE-2020-10543?s=alpine&n=perl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C5.30.3-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-10543?s=alpine&n=perl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C5.30.3-r0"
        }
      ]
    },
    {
      "id": "e3e4b8feed69405097fb3b268cc06472293049ec3ecbc172510268848e18ff77",
      "category": "container_scanning",
      "message": "CVE-2020-25659 on cryptography@2.9.2",
      "description": "RSA decryption was vulnerable to Bleichenbacher timing vulnerabilities, which would impact people using RSA decryption in online scenarios. This is fixed in cryptography 3.2. ",
      "cve": "CVE-2020-25659",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade cryptography@2.9.2 to 3.2",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/cryptography@2.9.2"
          },
          "version": "2.9.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-25659",
          "value": "CVE-2020-25659",
          "url": "https://scout.docker.com/v/CVE-2020-25659?s=github&n=cryptography&t=pypi&vr=%3C3.2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-25659?s=github&n=cryptography&t=pypi&vr=%3C3.2"
        }
      ]
    },
    {
      "id": "07e10e4e72a2670efd722ebfbc62ef97d96dedaf103af9168cf6f93e3fcb900b",
      "category": "container_scanning",
      "message": "CVE-2020-28972 on salt@2019.2.3",
      "description": "In SaltStack Salt before 3002.5, authentication to VMware vcenter, vsphere, and esxi servers (in the `vmware.py` files) does not always validate the SSL/TLS certificate.",
      "cve": "CVE-2020-28972",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-28972",
          "value": "CVE-2020-28972",
          "url": "https://scout.docker.com/v/CVE-2020-28972?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-28972?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ]
    },
    {
      "id": "27ee4463f3d05a6370f727156ba9381901ad6c21be3aff7929400df4cf9737f1",
      "category": "container_scanning",
      "message": "CVE-2020-10878 on alpine/perl@5.30.1-r0",
      "cve": "CVE-2020-10878",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/perl@5.30.1-r0 to 5.30.3-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/perl@5.30.1-r0?os_name=alpine&os_version=3.11"
          },
          "version": "5.30.1-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-10878",
          "value": "CVE-2020-10878",
          "url": "https://scout.docker.com/v/CVE-2020-10878?s=alpine&n=perl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C5.30.3-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-10878?s=alpine&n=perl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C5.30.3-r0"
        }
      ]
    },
    {
      "id": "8e144da9a112a3da7ab6da934024076fd7d3f637113e464b0179999d851cc84c",
      "category": "container_scanning",
      "message": "CVE-2018-6594 on pycrypto@2.6.1",
      "description": "lib/Crypto/PublicKey/ElGamal.py in PyCrypto through 2.6.1 generates weak ElGamal key parameters, which allows attackers to obtain sensitive information by reading ciphertext data (i.e., it does not have semantic security in face of a ciphertext-only attack). The Decisional Diffie-Hellman (DDH) assumption does not hold for PyCrypto's ElGamal implementation.",
      "cve": "CVE-2018-6594",
      "severity": "High",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/pycrypto@2.6.1"
          },
          "version": "2.6.1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2018-6594",
          "value": "CVE-2018-6594",
          "url": "https://scout.docker.com/v/CVE-2018-6594?s=github&n=pycrypto&t=pypi&vr=%3C%3D2.6.1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2018-6594?s=github&n=pycrypto&t=pypi&vr=%3C%3D2.6.1"
        }
      ]
    },
    {
      "id": "e93b37ca3b9124a779dbcbdfdb69335490b2bd7b7cb18c09578ae4e29e3a022b",
      "category": "container_scanning",
      "message": "CVE-2021-33503 on urllib3@1.25.9",
      "description": "### Impact\n\nWhen provided with a URL containing many `@` characters in the authority component the authority regular expression exhibits catastrophic backtracking causing a denial of service if a URL were passed as a parameter or redirected to via an HTTP redirect.\n\n\n### Patches\n\nThe issue has been fixed in urllib3 v1.26.5.\n\n### References\n\n- [CVE-2021-33503](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-33503)\n- [JVNVU#92413403 (English)](https://jvn.jp/en/vu/JVNVU92413403/)\n- [JVNVU#92413403 (Japanese)](https://jvn.jp/vu/JVNVU92413403/)\n- [urllib3 v1.26.5](https://github.com/urllib3/urllib3/releases/tag/1.26.5)\n\n### For more information\nIf you have any questions or comments about this advisory:\n* Ask in our [community Discord](https://discord.gg/urllib3)\n* Email [sethmichaellarson@gmail.com](mailto:sethmichaellarson@gmail.com)\n",
      "cve": "CVE-2021-33503",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade urllib3@1.25.9 to 1.26.5",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/urllib3@1.25.9"
          },
          "version": "1.25.9"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-33503",
          "value": "CVE-2021-33503",
          "url": "https://scout.docker.com/v/CVE-2021-33503?s=github&n=urllib3&t=pypi&vr=%3E%3D1.25.4%2C%3C1.26.5"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-33503?s=github&n=urllib3&t=pypi&vr=%3E%3D1.25.4%2C%3C1.26.5"
        }
      ]
    },
    {
      "id": "4c2b5da6446c3ef930f3b49c652f44de9c7a91f553694c84bb387dcc243d6fe5",
      "category": "container_scanning",
      "message": "CVE-2022-22934 on salt@2019.2.3",
      "description": "An issue was discovered in SaltStack Salt in versions before 3002.8, 3003.4, 3004.1. Salt Masters do not sign pillar data with the minion’s public key, which can result in attackers substituting arbitrary pillar data.",
      "cve": "CVE-2022-22934",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3002.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2022-22934",
          "value": "CVE-2022-22934",
          "url": "https://scout.docker.com/v/CVE-2022-22934?s=github&n=salt&t=pypi&vr=%3C3002.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2022-22934?s=github&n=salt&t=pypi&vr=%3C3002.8"
        }
      ]
    },
    {
      "id": "3f28a7e7f229ec9dc3d9da02014400ad41722b90ac8894afca93695d4744e73f",
      "category": "container_scanning",
      "message": "CVE-2022-40897 on setuptools@46.1.3",
      "description": "Python Packaging Authority (PyPA)'s setuptools is a library designed to facilitate packaging Python projects. Setuptools version 65.5.0 and earlier could allow remote attackers to cause a denial of service by fetching malicious HTML from a PyPI package or custom PackageIndex page due to a vulnerable Regular Expression in `package_index`. This has been patched in version 65.5.1.",
      "cve": "CVE-2022-40897",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade setuptools@46.1.3 to 65.5.1",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/setuptools@46.1.3"
          },
          "version": "46.1.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2022-40897",
          "value": "CVE-2022-40897",
          "url": "https://scout.docker.com/v/CVE-2022-40897?s=github&n=setuptools&t=pypi&vr=%3C65.5.1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2022-40897?s=github&n=setuptools&t=pypi&vr=%3C65.5.1"
        }
      ]
    },
    {
      "id": "12ad5e2e33130fe6b5ec1ff0b30624f1243252ba894c3180526381e272b4033e",
      "category": "container_scanning",
      "message": "CVE-2023-50782 on cryptography@2.9.2",
      "description": "A flaw was found in the python-cryptography package. This issue may allow a remote attacker to decrypt captured messages in TLS servers that use RSA key exchanges, which may lead to exposure of confidential or sensitive data.",
      "cve": "CVE-2023-50782",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade cryptography@2.9.2 to 42.0.0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/cryptography@2.9.2"
          },
          "version": "2.9.2"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2023-50782",
          "value": "CVE-2023-50782",
          "url": "https://scout.docker.com/v/CVE-2023-50782?s=github&n=cryptography&t=pypi&vr=%3C42.0.0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2023-50782?s=github&n=cryptography&t=pypi&vr=%3C42.0.0"
        }
      ]
    },
    {
      "id": "7d37eabfd7fbd4a41d8e75d3534ab45f39a02c6cec6045429d73ecad1ba300f0",
      "category": "container_scanning",
      "message": "CVE-2021-25282 on salt@2019.2.3",
      "description": "An issue was discovered in through SaltStack Salt before 3002.5. The `salt.wheel.pillar_roots.write` method is vulnerable to directory traversal.",
      "cve": "CVE-2021-25282",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-25282",
          "value": "CVE-2021-25282",
          "url": "https://scout.docker.com/v/CVE-2021-25282?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-25282?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ]
    },
    {
      "id": "fbfdedf6ecc166c160d9e2d9877671df490c66aff0145a0aa94ee9c2952bf3d6",
      "category": "container_scanning",
      "message": "CVE-2021-39537 on alpine/ncurses@6.1_p20200118-r3",
      "cve": "CVE-2021-39537",
      "severity": "High",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/ncurses@6.1_p20200118-r3?os_name=alpine&os_version=3.11"
          },
          "version": "6.1_p20200118-r3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-39537",
          "value": "CVE-2021-39537",
          "url": "https://scout.docker.com/v/CVE-2021-39537?s=alpine&n=ncurses&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D6.1_p20200118-r4"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-39537?s=alpine&n=ncurses&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D6.1_p20200118-r4"
        }
      ]
    },
    {
      "id": "4ff280b6325b32721dc9704eaf3240ccb4027cc9c1da5df8a1b8911257022906",
      "category": "container_scanning",
      "message": "CVE-2022-22936 on salt@2019.2.3",
      "description": "An issue was discovered in SaltStack Salt in versions before 3002.8, 3003.4, 3004.1. Job publishes and file server replies are susceptible to replay attacks, which can result in an attacker replaying job publishes causing minions to run old jobs. File server replies can also be re-played. A sufficient craft attacker could gain root access on minion under certain scenarios.",
      "cve": "CVE-2022-22936",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3002.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2022-22936",
          "value": "CVE-2022-22936",
          "url": "https://scout.docker.com/v/CVE-2022-22936?s=github&n=salt&t=pypi&vr=%3C3002.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2022-22936?s=github&n=salt&t=pypi&vr=%3C3002.8"
        }
      ]
    },
    {
      "id": "b9ac66afa80e5b8f1bdf7f79cab295a66402aedb601488e331888e39e191f844",
      "category": "container_scanning",
      "message": "CVE-2022-22941 on salt@2019.2.3",
      "description": "An issue was discovered in SaltStack Salt in versions before 3002.8, 3003.4, 3004.1. When configured as a Master-of-Masters, with a publisher_acl, if a user configured in the publisher_acl targets any minion connected to the Syndic, the Salt Master incorrectly interpreted no valid targets as valid, allowing configured users to target any of the minions connected to the syndic with their configured commands. This requires a syndic master combined with publisher_acl configured on the Master-of-Masters, allowing users specified in the publisher_acl to bypass permissions, publishing authorized commands to any configured minion.",
      "cve": "CVE-2022-22941",
      "severity": "High",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3002.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2022-22941",
          "value": "CVE-2022-22941",
          "url": "https://scout.docker.com/v/CVE-2022-22941?s=github&n=salt&t=pypi&vr=%3C3002.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2022-22941?s=github&n=salt&t=pypi&vr=%3C3002.8"
        }
      ]
    },
    {
      "id": "b5bdf457d2594efc1939780571356f7a9d088d58a1e5aa236c4f58745fc58ea3",
      "category": "container_scanning",
      "message": "CVE-2021-3144 on salt@2019.2.3",
      "description": "In SaltStack Salt before 3002.5, eauth tokens can be used once after expiration. (They might be used to run command against the salt master or minions.)",
      "cve": "CVE-2021-3144",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-3144",
          "value": "CVE-2021-3144",
          "url": "https://scout.docker.com/v/CVE-2021-3144?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-3144?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ]
    },
    {
      "id": "bd8c1db9b89cc1566dee893fc0a9fe903ef9a2bfa5215dfd439b4aaa4cb547d8",
      "category": "container_scanning",
      "message": "CVE-2021-36159 on alpine/apk-tools@2.10.5-r0",
      "cve": "CVE-2021-36159",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/apk-tools@2.10.5-r0 to 2.10.7-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/apk-tools@2.10.5-r0?os_name=alpine&os_version=3.11"
          },
          "version": "2.10.5-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-36159",
          "value": "CVE-2021-36159",
          "url": "https://scout.docker.com/v/CVE-2021-36159?s=alpine&n=apk-tools&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C2.10.7-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-36159?s=alpine&n=apk-tools&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C2.10.7-r0"
        }
      ]
    },
    {
      "id": "86481e8080bcaa918d23e55c729380385d78d259b75d47c165de72c7713abf0c",
      "category": "container_scanning",
      "message": "CVE-2013-7459 on pycrypto@2.6.1",
      "description": "Heap-based buffer overflow in the ALGnew function in block_templace.c in Python Cryptography Toolkit (aka pycrypto) allows remote attackers to execute arbitrary code as demonstrated by a crafted iv parameter to cryptmsg.py.",
      "cve": "CVE-2013-7459",
      "severity": "Critical",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/pycrypto@2.6.1"
          },
          "version": "2.6.1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2013-7459",
          "value": "CVE-2013-7459",
          "url": "https://scout.docker.com/v/CVE-2013-7459?s=github&n=pycrypto&t=pypi&vr=%3C%3D2.6.1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2013-7459?s=github&n=pycrypto&t=pypi&vr=%3C%3D2.6.1"
        }
      ]
    },
    {
      "id": "4cb5db20fc203f43aaeb705650c01e5d4ff0f375e7fe93647182d218a8a11b34",
      "category": "container_scanning",
      "message": "CVE-2020-11651 on salt@2019.2.3",
      "description": "An issue was discovered in SaltStack Salt before 2019.2.4 and 3000 before 3000.2. The salt-master process ClearFuncs class does not properly validate method calls. This allows a remote user to access some methods without authentication. These methods can be used to retrieve user tokens from the salt master and/or run arbitrary commands on salt minions.",
      "cve": "CVE-2020-11651",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.4",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-11651",
          "value": "CVE-2020-11651",
          "url": "https://scout.docker.com/v/CVE-2020-11651?s=github&n=salt&t=pypi&vr=%3C2019.2.4"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-11651?s=github&n=salt&t=pypi&vr=%3C2019.2.4"
        }
      ]
    },
    {
      "id": "e0812b765e3c09cabea5704b3266da0109cd39e5e42012fae3227ee9147bfc40",
      "category": "container_scanning",
      "message": "CVE-2020-14343 on pyyaml@3.13",
      "description": "A vulnerability was discovered in the PyYAML library in versions before 5.4, where it is susceptible to arbitrary code execution when it processes untrusted YAML files through the full_load method or with the FullLoader loader. Applications that use the library to process untrusted input may be vulnerable to this flaw. This flaw allows an attacker to execute arbitrary code on the system by abusing the python/object/new constructor. This flaw is due to an incomplete fix for CVE-2020-1747.",
      "cve": "CVE-2020-14343",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade pyyaml@3.13 to 5.4",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/pyyaml@3.13"
          },
          "version": "3.13"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-14343",
          "value": "CVE-2020-14343",
          "url": "https://scout.docker.com/v/CVE-2020-14343?s=github&n=pyyaml&t=pypi&vr=%3C5.4"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-14343?s=github&n=pyyaml&t=pypi&vr=%3C5.4"
        }
      ]
    },
    {
      "id": "5156018124811273cd7850daed0f710b96f3e68b5e90ab993643e4a6a800f3a6",
      "category": "container_scanning",
      "message": "CVE-2017-18342 on pyyaml@3.13",
      "description": "In PyYAML before 4.1, the `yaml.load()` API could execute arbitrary code. In other words, `yaml.safe_load` is not used.",
      "cve": "CVE-2017-18342",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade pyyaml@3.13 to 4.1",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/pyyaml@3.13"
          },
          "version": "3.13"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2017-18342",
          "value": "CVE-2017-18342",
          "url": "https://scout.docker.com/v/CVE-2017-18342?s=github&n=pyyaml&t=pypi&vr=%3C4.1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2017-18342?s=github&n=pyyaml&t=pypi&vr=%3C4.1"
        }
      ]
    },
    {
      "id": "40481524674cf61a956072beb6baf5d3e918dd87cd9823c49230aaa87595546a",
      "category": "container_scanning",
      "message": "CVE-2019-19317 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2019-19317",
      "severity": "Critical",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-19317",
          "value": "CVE-2019-19317",
          "url": "https://scout.docker.com/v/CVE-2019-19317?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-19317?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "893077d984d5a639d3bf2b42d3da5f7876806f51cdbd2b370c01e95a6ab0646a",
      "category": "container_scanning",
      "message": "CVE-2019-19646 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2019-19646",
      "severity": "Critical",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2019-19646",
          "value": "CVE-2019-19646",
          "url": "https://scout.docker.com/v/CVE-2019-19646?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2019-19646?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "026e5777660fa9186f714e584db0beaa5f3e8db4e1a2987511047825b64dbc07",
      "category": "container_scanning",
      "message": "CVE-2020-11656 on alpine/sqlite@3.30.1-r1",
      "cve": "CVE-2020-11656",
      "severity": "Critical",
      "confidence": "Unknown",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/sqlite@3.30.1-r1?os_name=alpine&os_version=3.11"
          },
          "version": "3.30.1-r1"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-11656",
          "value": "CVE-2020-11656",
          "url": "https://scout.docker.com/v/CVE-2020-11656?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-11656?s=alpine&n=sqlite&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C%3D3.30.1-r2"
        }
      ]
    },
    {
      "id": "6f85646335031a9f8cc650a88766695fa7e99e2f716459f57b4a295af144b6c2",
      "category": "container_scanning",
      "message": "CVE-2020-16846 on salt@2019.2.3",
      "description": "An issue was discovered in SaltStack Salt through 3002. Sending crafted web requests to the Salt API, with the SSH client enabled, can result in shell injection.",
      "cve": "CVE-2020-16846",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.6",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-16846",
          "value": "CVE-2020-16846",
          "url": "https://scout.docker.com/v/CVE-2020-16846?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.6"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-16846?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.6"
        }
      ]
    },
    {
      "id": "40d5cbb416bdbe8e788dc82a80dfff64c8981fd4bf1b1c6e0a455e79be43919e",
      "category": "container_scanning",
      "message": "CVE-2020-25592 on salt@2019.2.3",
      "description": "In SaltStack the salt-netapi improperly validates eauth credentials and tokens. A user can bypass authentication and invoke Salt SSH.",
      "cve": "CVE-2020-25592",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.7",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2020-25592",
          "value": "CVE-2020-25592",
          "url": "https://scout.docker.com/v/CVE-2020-25592?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.7"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2020-25592?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.7"
        }
      ]
    },
    {
      "id": "2e4a4ef824ecaf53005f88f14168d7a33751df8ade007b3b0ed1f37d317884ec",
      "category": "container_scanning",
      "message": "CVE-2021-25281 on salt@2019.2.3",
      "description": "An issue was discovered in through SaltStack Salt before 3002.5. salt-api does not honor eauth credentials for the wheel_async client. Thus, an attacker can remotely run any wheel modules on the master.",
      "cve": "CVE-2021-25281",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-25281",
          "value": "CVE-2021-25281",
          "url": "https://scout.docker.com/v/CVE-2021-25281?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-25281?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ]
    },
    {
      "id": "dd5140f725f092db6f5aad65578c56ee530263a381b7a31b901972f17d7f99cd",
      "category": "container_scanning",
      "message": "CVE-2021-25283 on salt@2019.2.3",
      "description": "An issue was discovered in through SaltStack Salt before 3002.5. The jinja renderer does not protect against server side template injection attacks.",
      "cve": "CVE-2021-25283",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-25283",
          "value": "CVE-2021-25283",
          "url": "https://scout.docker.com/v/CVE-2021-25283?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-25283?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ]
    },
    {
      "id": "23cef529e2083e0799fac58c77232cd81354e1213cd60971670bb20d7d94486d",
      "category": "container_scanning",
      "message": "CVE-2021-3148 on salt@2019.2.3",
      "description": "An issue was discovered in SaltStack Salt before 3002.5. Sending crafted web requests to the Salt API can result in `salt.utils.thin.gen_thin()` command injection because of different handling of single versus double quotes. This is related to `salt/utils/thin.py`.",
      "cve": "CVE-2021-3148",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-3148",
          "value": "CVE-2021-3148",
          "url": "https://scout.docker.com/v/CVE-2021-3148?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-3148?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ]
    },
    {
      "id": "6a403fb054d0889b67e0052a79b1ceb1576d7fb7243a846b0c8a20fb64661bd1",
      "category": "container_scanning",
      "message": "CVE-2021-3197 on salt@2019.2.3",
      "description": "An issue was discovered in SaltStack Salt before 3002.5. The salt-api's ssh client is vulnerable to a shell injection by including ProxyCommand in an argument, or via ssh_options provided in an API request.",
      "cve": "CVE-2021-3197",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 2019.2.8",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-3197",
          "value": "CVE-2021-3197",
          "url": "https://scout.docker.com/v/CVE-2021-3197?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-3197?s=github&n=salt&t=pypi&vr=%3E%3D2019.2.0%2C%3C2019.2.8"
        }
      ]
    },
    {
      "id": "3d11e5c7b759aa458586f0c4100c5ccb8628963f509e37062627d23868144b4c",
      "category": "container_scanning",
      "message": "CVE-2021-33226 on salt@2019.2.3",
      "description": "Buffer Overflow vulnerability in Saltstack v.3003 and before allows attacker to execute arbitrary code via the func variable in salt/salt/modules/status.py file.",
      "cve": "CVE-2021-33226",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade salt@2019.2.3 to 3003.1",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:pypi/salt@2019.2.3"
          },
          "version": "2019.2.3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-33226",
          "value": "CVE-2021-33226",
          "url": "https://scout.docker.com/v/CVE-2021-33226?s=pypa&n=salt&t=pypi&vr=%3C3003.1"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-33226?s=pypa&n=salt&t=pypi&vr=%3C3003.1"
        }
      ]
    },
    {
      "id": "b905e87744321ddc629bf113353875662bc8b48c6a169b21affe1fd0d3ed41e7",
      "category": "container_scanning",
      "message": "CVE-2021-3711 on alpine/openssl@1.1.1g-r0",
      "cve": "CVE-2021-3711",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/openssl@1.1.1g-r0 to 1.1.1l-r0",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/openssl@1.1.1g-r0?os_name=alpine&os_version=3.11"
          },
          "version": "1.1.1g-r0"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2021-3711",
          "value": "CVE-2021-3711",
          "url": "https://scout.docker.com/v/CVE-2021-3711?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1l-r0"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2021-3711?s=alpine&n=openssl&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.1.1l-r0"
        }
      ]
    },
    {
      "id": "7615843865717e6d494460f9dc2e00f704594823c4b8488abd4196d80fbf3aee",
      "category": "container_scanning",
      "message": "CVE-2022-37434 on alpine/zlib@1.2.11-r3",
      "cve": "CVE-2022-37434",
      "severity": "Critical",
      "confidence": "Unknown",
      "solution": "Upgrade alpine/zlib@1.2.11-r3 to 1.2.11-r4",
      "scanner": {
        "id": "docker-scout",
        "name": "Docker Scout"
      },
      "location": {
        "dependency": {
          "package": {
            "name": "pkg:apk/alpine/zlib@1.2.11-r3?os_name=alpine&os_version=3.11"
          },
          "version": "1.2.11-r3"
        },
        "operating_system": "alpine 3.11",
        "image": "saltstack/salt:2019.2.3"
      },
      "identifiers": [
        {
          "type": "cve",
          "name": "CVE-2022-37434",
          "value": "CVE-2022-37434",
          "url": "https://scout.docker.com/v/CVE-2022-37434?s=alpine&n=zlib&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.2.11-r4"
        }
      ],
      "links": [
        {
          "url": "https://scout.docker.com/v/CVE-2022-37434?s=alpine&n=zlib&ns=alpine&t=apk&osn=alpine&osv=3.11&vr=%3C1.2.11-r4"
        }
      ]
    }
  ],
  "remediations": [],
  "scan": {
    "start_time": "2025-10-23T01:17:49",
    "end_time": "2025-10-23T01:17:49",
    "status": "success",
    "type": "container_scanning",
    "scanner": {
      "id": "docker-scout",
      "name": "Docker Scout",
      "version": "1.18.3",
      "vendor": {
        "name": "Docker"
      }
    },
    "analyzer": {
      "id": "docker-scout",
      "name": "Docker Scout",
      "version": "1.18.3",
      "vendor": {
        "name": "Docker"
      }
    }
  }
}
